From 94bef9aaad01c40dfa194b9292a865cb1e49382c Mon Sep 17 00:00:00 2001
From: John Vincent <john.vincent.xa.bp.renesas.com>
Date: Fri, 25 Aug 2023 10:32:27 +0100
Subject: [PATCH] smarc rzg2l device tree to match linux upstream

u-boot device tree to match with linux 6.4.3
clock, pfc and gpio updates

Signed-off-by: John Vincent <john.vincent.xa.bp.renesas.com>
---
 arch/arm/dts/r9a07g044l.dtsi                | 1117 +++++++++++++------
 arch/arm/dts/smarc-rzg2l.dts                |  151 +--
 drivers/clk/renesas/clk-rzg2l.c             |   19 +-
 drivers/clk/renesas/r9a07g044l-cpg-mssr.c   |  336 +++---
 drivers/net/ravb.c                          |    1 +
 drivers/phy/phy-rcar-gen3.c                 |    1 +
 drivers/pinctrl/renesas/pfc-rzg2l.c         |    4 +-
 drivers/power/regulator/regulator_common.c  |    5 +-
 drivers/serial/serial_sh.c                  |    1 +
 drivers/watchdog/rzg2l_wdt.c                |    1 +
 include/dt-bindings/clock/r9a07g044l-cpg.h  |  292 +++--
 include/dt-bindings/pinctrl/rzg2l-pinctrl.h |   19 +-
 12 files changed, 1233 insertions(+), 714 deletions(-)

diff --git a/arch/arm/dts/r9a07g044l.dtsi b/arch/arm/dts/r9a07g044l.dtsi
index 57b0fdec25..ab9154262c 100644
--- a/arch/arm/dts/r9a07g044l.dtsi
+++ b/arch/arm/dts/r9a07g044l.dtsi
@@ -9,7 +9,7 @@
 #include <dt-bindings/clock/r9a07g044l-cpg.h>
 
 / {
-	compatible = "renesas,r9a07g044l";
+	compatible = "renesas,r9a07g044l","renesas,r9a07g044";
 	#address-cells = <2>;
 	#size-cells = <2>;
 
@@ -23,25 +23,120 @@
 		spi2 = &rspi2;
 	};
 
-	/* Bootloader run on 1 core, just delare 1 core in dts but 
+	cluster0_opp: opp-table-0 {
+		compatible = "operating-points-v2";
+		opp-shared;
+
+		opp-150000000 {
+			opp-hz = /bits/ 64 <150000000>;
+			opp-microvolt = <1100000>;
+			clock-latency-ns = <300000>;
+		};
+		opp-300000000 {
+			opp-hz = /bits/ 64 <300000000>;
+			opp-microvolt = <1100000>;
+			clock-latency-ns = <300000>;
+		};
+		opp-600000000 {
+			opp-hz = /bits/ 64 <600000000>;
+			opp-microvolt = <1100000>;
+			clock-latency-ns = <300000>;
+		};
+		opp-1200000000 {
+			opp-hz = /bits/ 64 <1200000000>;
+			opp-microvolt = <1100000>;
+			clock-latency-ns = <300000>;
+			opp-suspend;
+		};
+	};
+
+	/* Bootloader run on 1 core, just delare 1 core in dts but
 	the dtb can support multicore device */
 	cpus {
-		#address-cells = <0x2>;
-		#size-cells = <0x0>;
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		cpu-map {
+			cluster0 {
+				core0 {
+					cpu = <&cpu0>;
+				};
+				core1 {
+					cpu = <&cpu1>;
+				};
+			};
+		};
 
 		cpu0: cpu@0 {
+			compatible = "arm,cortex-a55";
+			reg = <0>;
 			device_type = "cpu";
+			#cooling-cells = <2>;
+			next-level-cache = <&L3_CA55>;
+			enable-method = "psci";
+			clocks = <&cpg CPG_CORE R9A07G044_CLK_I>;
+			operating-points-v2 = <&cluster0_opp>;
+		};
+
+		cpu1: cpu@100 {
 			compatible = "arm,cortex-a55";
-			reg = <0x0 0x0>;
+			reg = <0x100>;
+			device_type = "cpu";
+			next-level-cache = <&L3_CA55>;
 			enable-method = "psci";
-			next-level-cache = <&l3>;
-			#cooling-cells = <2>;
+			clocks = <&cpg CPG_CORE R9A07G044_CLK_I>;
+			operating-points-v2 = <&cluster0_opp>;
 		};
 
-		l3: l3-cache {
+		L3_CA55: cache-controller-0 {
 			compatible = "cache";
 			cache-unified;
-			cache-level = <3>;
+			cache-size = <0x40000>;
+		};
+	};
+
+	gpu_opp_table: opp-table-1 {
+		compatible = "operating-points-v2";
+
+		opp-500000000 {
+			opp-hz = /bits/ 64 <500000000>;
+			opp-microvolt = <1100000>;
+		};
+
+		opp-400000000 {
+			opp-hz = /bits/ 64 <400000000>;
+			opp-microvolt = <1100000>;
+		};
+
+		opp-250000000 {
+			opp-hz = /bits/ 64 <250000000>;
+			opp-microvolt = <1100000>;
+		};
+
+		opp-200000000 {
+			opp-hz = /bits/ 64 <200000000>;
+			opp-microvolt = <1100000>;
+		};
+
+
+		opp-125000000 {
+			opp-hz = /bits/ 64 <125000000>;
+			opp-microvolt = <1100000>;
+		};
+
+		opp-100000000 {
+			opp-hz = /bits/ 64 <100000000>;
+			opp-microvolt = <1100000>;
+		};
+
+		opp-62500000 {
+			opp-hz = /bits/ 64 <62500000>;
+			opp-microvolt = <1100000>;
+		};
+
+		opp-50000000 {
+			opp-hz = /bits/ 64 <50000000>;
+			opp-microvolt = <1100000>;
 		};
 	};
 
@@ -58,10 +153,17 @@
 	audio_clk1: audio_clk1 {
 		compatible = "fixed-clock";
 		#clock-cells = <0>;
-		clock-frequency = <0>;
+		clock-frequency = <0xac4400>;
 	};
 
 	audio_clk2: audio_clk2 {
+		compatible = "fixed-clock";
+		#clock-cells = <0>;
+		clock-frequency = <0xbb8000>;
+	};
+
+	/* External CAN clock - to be overridden by boards that provide it */
+	can_clk: can-clk {
 		compatible = "fixed-clock";
 		#clock-cells = <0>;
 		clock-frequency = <0>;
@@ -74,6 +176,13 @@
 		clock-frequency = <0>;
 	};
 
+	firmware {
+
+		optee {
+			compatible = "linaro,optee-tz";
+			method = "smc";
+		};
+	};
 	soc: soc {
 		compatible = "simple-bus";
 		interrupt-parent = <&gic>;
@@ -89,24 +198,24 @@
 			status = "disabled";
 		};
 
+
 		wdt1: watchdog@12800c00 {
-			compatible = "renesas,r9a07g044l-wdt";
+			compatible = "renesas,r9a07g044l-wdt",
+					"renesas,r9a07g044-wdt","renesas,rzg2l-wdt";
 			reg = <0 0x12800C00 0 0x400>;
-			clocks = <&xinclk>;
-			power-domains = <&cpg>;
-			status = "disabled";
-		};
-
-		wdt2: watchdog@12800400 {
-			compatible = "renesas,r9a07g044l-wdt";
-			reg = <0 0x12800400 0 0x400>;
-			clocks = <&xinclk>;
+			clocks = <&cpg CPG_MOD R9A07G044_WDT1_PCLK>,
+				 <&cpg CPG_MOD R9A07G044_WDT1_CLK>;
+			clock-names = "pclk", "oscclk";
+			interrupts = <GIC_SPI 51 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 52 IRQ_TYPE_LEVEL_HIGH>;
+			interrupt-names = "wdt", "perrout";
+			resets = <&cpg R9A07G044_WDT1_PRESETN>;
 			power-domains = <&cpg>;
 			status = "disabled";
 		};
 
 		cpg: clock-controller@11010000 {
-			compatible = "renesas,r9a07g044l-cpg";
+			compatible = "renesas,r9a07g044l-cpg","renesas,r9a07g044-cpg";
 			reg = <0 0x11010000 0 0x10000>;
 			clocks = <&xinclk>;
 			clock-names = "xinclk";
@@ -115,13 +224,74 @@
 			#power-domain-cells = <0>;
 		};
 
-		pinctrl: pin-controller@11030000 {
-			compatible = "renesas,r9a07g044l-pinctrl";
-			reg = <0 0x11030000 0 0x10000>;
+		irqc: interrupt-controller@110a0000 {
+			compatible = "renesas,r9a07g044-irqc",
+				     "renesas,rzg2l-irqc";
+			#interrupt-cells = <2>;
+			interrupt-controller;
+			reg = <0x0 0x110A0000 0 0x10000>;
+			interrupts = <GIC_SPI 1 IRQ_TYPE_LEVEL_HIGH>,
+				<GIC_SPI 2 IRQ_TYPE_LEVEL_HIGH>,
+				<GIC_SPI 3 IRQ_TYPE_LEVEL_HIGH>,
+				<GIC_SPI 4 IRQ_TYPE_LEVEL_HIGH>,
+				<GIC_SPI 5 IRQ_TYPE_LEVEL_HIGH>,
+				<GIC_SPI 6 IRQ_TYPE_LEVEL_HIGH>,
+				<GIC_SPI 7 IRQ_TYPE_LEVEL_HIGH>,
+				<GIC_SPI 8 IRQ_TYPE_LEVEL_HIGH>,
+				<GIC_SPI 0 IRQ_TYPE_LEVEL_HIGH>;
+			interrupt-names = "irq0", "irq1", "irq2", "irq3",
+				"irq4", "irq5", "irq6", "irq7", "nmi";
+			clocks = <&cpg CPG_MOD R9A07G044_IA55_CLK>;
+			resets = <&cpg R9A07G044_IA55_RESETN>;
+			power-domains = <&cpg>;
+			status = "disabled";
+		};
+
+		pinctrl: pinctrl@11030000 {
+			compatible = "renesas,r9a07g044l-pinctrl","renesas,r9a07g044-pinctrl";
+			reg = <0 0x11030000 0 0x10000>, <0 0x110a0020 0 0x30>;
 			gpio-controller;
 			#gpio-cells = <2>;
 			gpio-ranges = <&pinctrl 0 0 392>;
-			clocks = <&cpg CPG_MOD R9A07G044L_CLK_GPIO>;
+			clocks = <&cpg CPG_MOD R9A07G044_GPIO_HCLK>;
+			power-domains = <&cpg>;
+			resets = <&cpg R9A07G044_GPIO_RSTN>,
+				 <&cpg R9A07G044_GPIO_PORT_RESETN>,
+				 <&cpg R9A07G044_GPIO_SPARE_RESETN>;
+			interrupt-controller;
+			#interrupt-cells = <2>;
+			interrupts = <GIC_SPI 444 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 445 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 446 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 447 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 448 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 449 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 450 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 451 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 452 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 453 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 454 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 455 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 456 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 457 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 458 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 459 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 460 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 461 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 462 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 463 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 464 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 465 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 466 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 467 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 468 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 469 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 470 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 471 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 472 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 473 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 474 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 475 IRQ_TYPE_LEVEL_HIGH>;
 		};
 
 		gic: interrupt-controller@11900000 {
@@ -131,83 +301,166 @@
 			interrupt-controller;
 			reg = <0x0 0x11900000 0 0x40000>,
 			      <0x0 0x11940000 0 0x60000>;
-			clocks = <&cpg CPG_MOD R9A07G044L_CLK_GIC600>;
-			clock-names = "gic6000";
-			interrupts = <GIC_PPI 9
-				(GIC_CPU_MASK_SIMPLE(2) | IRQ_TYPE_LEVEL_LOW)>;
+			interrupts = <GIC_PPI 9 IRQ_TYPE_LEVEL_LOW>;
 		};
 
 		scif0: serial@1004b800 {
-			compatible = "renesas,scif-r9a07g044l", "renesas,scif";
+			compatible = "renesas,scif-r9a07g044", "renesas,scif";
 			reg = <0 0x1004b800 0 0x400>;
-			interrupts =
-				<GIC_SPI 380 IRQ_TYPE_LEVEL_HIGH>,
-				<GIC_SPI 382 IRQ_TYPE_LEVEL_HIGH>,
-				<GIC_SPI 383 IRQ_TYPE_LEVEL_HIGH>,
-				<GIC_SPI 381 IRQ_TYPE_LEVEL_HIGH>,
-				<GIC_SPI 384 IRQ_TYPE_LEVEL_HIGH>,
-				<GIC_SPI 384 IRQ_TYPE_LEVEL_HIGH>;
-			clocks = <&cpg CPG_MOD R9A07G044L_CLK_SCIF0>;
+			interrupts = <GIC_SPI 380 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 382 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 383 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 381 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 384 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 384 IRQ_TYPE_LEVEL_HIGH>;
+			interrupt-names = "eri", "rxi", "txi",
+					  "bri", "dri", "tei";
+			clocks = <&cpg CPG_MOD R9A07G044_SCIF0_CLK_PCK>;
 			clock-names = "fck";
-			resets = <&cpg R9A07G044L_CLK_SCIF0>;
+			dmas = <&dmac 0x4e79>, <&dmac 0x4e7a>;
+			dma-names = "tx", "rx";
 			power-domains = <&cpg>;
-			status = "disable";
+			resets = <&cpg R9A07G044_SCIF0_RST_SYSTEM_N>;
+			status = "disabled";
 		};
 
 		scif1: serial@1004bc00 {
-			compatible = "renesas,scif-r9a07g044l", "renesas,scif";
+			compatible = "renesas,scif-r9a07g044", "renesas,scif";
 			reg = <0 0x1004bc00 0 0x400>;
-			interrupts =
-				<GIC_SPI 385 IRQ_TYPE_LEVEL_HIGH>,
-				<GIC_SPI 387 IRQ_TYPE_LEVEL_HIGH>,
-				<GIC_SPI 388 IRQ_TYPE_LEVEL_HIGH>,
-				<GIC_SPI 386 IRQ_TYPE_LEVEL_HIGH>,
-				<GIC_SPI 389 IRQ_TYPE_LEVEL_HIGH>,
-				<GIC_SPI 389 IRQ_TYPE_LEVEL_HIGH>;
-			clocks = <&cpg CPG_MOD R9A07G044L_CLK_SCIF1>;
+			interrupts = <GIC_SPI 385 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 387 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 388 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 386 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 389 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 389 IRQ_TYPE_LEVEL_HIGH>;
+			interrupt-names = "eri", "rxi", "txi",
+					  "bri", "dri", "tei";
+			clocks = <&cpg CPG_MOD R9A07G044_SCIF1_CLK_PCK>;
 			clock-names = "fck";
-			resets = <&cpg R9A07G044L_CLK_SCIF1>;
+			dmas = <&dmac 0x4e7d>, <&dmac 0x4e7e>;
+			dma-names = "tx", "rx";
 			power-domains = <&cpg>;
-			status = "disable";
+			resets = <&cpg R9A07G044_SCIF1_RST_SYSTEM_N>;
+			status = "disabled";
 		};
 
 		scif2: serial@1004c000 {
-			compatible = "renesas,scif-r9a07g044l", "renesas,scif";
+			compatible = "renesas,scif-r9a07g044", "renesas,scif";
 			reg = <0 0x1004c000 0 0x400>;
-			interrupts =
-				<GIC_SPI 390 IRQ_TYPE_LEVEL_HIGH>,
-				<GIC_SPI 392 IRQ_TYPE_LEVEL_HIGH>,
-				<GIC_SPI 393 IRQ_TYPE_LEVEL_HIGH>,
-				<GIC_SPI 391 IRQ_TYPE_LEVEL_HIGH>,
-				<GIC_SPI 394 IRQ_TYPE_LEVEL_HIGH>,
-				<GIC_SPI 394 IRQ_TYPE_LEVEL_HIGH>;
-			clocks = <&cpg CPG_MOD R9A07G044L_CLK_SCIF2>;
+			interrupts = <GIC_SPI 390 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 392 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 393 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 391 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 394 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 394 IRQ_TYPE_LEVEL_HIGH>;
+			interrupt-names = "eri", "rxi", "txi",
+					  "bri", "dri", "tei";
+			clocks = <&cpg CPG_MOD R9A07G044_SCIF2_CLK_PCK>;
+			clock-names = "fck";
+			dmas = <&dmac 0x4e81>, <&dmac 0x4e82>;
+			dma-names = "tx", "rx";
+			power-domains = <&cpg>;
+			resets = <&cpg R9A07G044_SCIF2_RST_SYSTEM_N>;
+			status = "disabled";
+		};
+
+		scif3: serial@1004c400 {
+			compatible = "renesas,scif-r9a07g044";
+			reg = <0 0x1004c400 0 0x400>;
+			interrupts = <GIC_SPI 395 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 397 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 398 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 396 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 399 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 399 IRQ_TYPE_LEVEL_HIGH>;
+			interrupt-names = "eri", "rxi", "txi",
+					  "bri", "dri", "tei";
+			clocks = <&cpg CPG_MOD R9A07G044_SCIF3_CLK_PCK>;
 			clock-names = "fck";
-			resets = <&cpg R9A07G044L_CLK_SCIF2>;
+			dmas = <&dmac 0x4e85>, <&dmac 0x4e86>;
+			dma-names = "tx", "rx";
 			power-domains = <&cpg>;
-			status = "disable";
+			resets = <&cpg R9A07G044_SCIF3_RST_SYSTEM_N>;
+			status = "disabled";
 		};
 
+		scif4: serial@1004c800 {
+			compatible = "renesas,scif-r9a07g044";
+			reg = <0 0x1004c800 0 0x400>;
+			interrupts = <GIC_SPI 400 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 402 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 403 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 401 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 404 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 404 IRQ_TYPE_LEVEL_HIGH>;
+			interrupt-names = "eri", "rxi", "txi",
+					  "bri", "dri", "tei";
+			clocks = <&cpg CPG_MOD R9A07G044_SCIF4_CLK_PCK>;
+			clock-names = "fck";
+			dmas = <&dmac 0x4e89>, <&dmac 0x4e8a>;
+			dma-names = "tx", "rx";
+			power-domains = <&cpg>;
+			resets = <&cpg R9A07G044_SCIF4_RST_SYSTEM_N>;
+			status = "disabled";
+		};
+		sci0: serial@1004d000 {
+			compatible = "renesas,r9a07g044-sci", "renesas,sci","renesas,rz-sci";
+			reg = <0 0x1004d000 0 0x400>;
+			interrupts = <GIC_SPI 405 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 406 IRQ_TYPE_EDGE_RISING>,
+				     <GIC_SPI 407 IRQ_TYPE_EDGE_RISING>,
+				     <GIC_SPI 408 IRQ_TYPE_LEVEL_HIGH>;
+			interrupt-names = "eri", "rxi", "txi", "tei";
+			clocks = <&cpg CPG_MOD R9A07G044_SCI0_CLKP>;
+			clock-names = "fck";
+			power-domains = <&cpg>;
+			resets = <&cpg R9A07G044_SCI0_RST>;
+			status = "disabled";
+		};
+
+		sci1: serial@1004d400 {
+			compatible = "renesas,r9a07g044-sci", "renesas,sci", "renesas,rz-sci";
+			reg = <0 0x1004d400 0 0x400>;
+			interrupts = <GIC_SPI 409 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 410 IRQ_TYPE_EDGE_RISING>,
+				     <GIC_SPI 411 IRQ_TYPE_EDGE_RISING>,
+				     <GIC_SPI 412 IRQ_TYPE_LEVEL_HIGH>;
+			interrupt-names = "eri", "rxi", "txi", "tei";
+			clocks = <&cpg CPG_MOD R9A07G044_SCI1_CLKP>;
+			clock-names = "fck";
+			power-domains = <&cpg>;
+			resets = <&cpg R9A07G044_SCI1_RST>;
+			status = "disabled";
+		};
 		sdhi0: sd@11c00000 {
-			compatible = "renesas,sdhi-r9a07g044l",
+			compatible = "renesas,sdhi-r9a07g044l","renesas,sdhi-r9a07g044",
 				     "renesas,rcar-gen3-sdhi";
-			reg = <0 0x11c00000 0 0x10000>;
+			reg = <0x0 0x11c00000 0 0x10000>;
 			interrupts = <GIC_SPI 104 IRQ_TYPE_LEVEL_HIGH>,
 				     <GIC_SPI 105 IRQ_TYPE_LEVEL_HIGH>;
-			clocks = <&cpg CPG_MOD R9A07G044L_CLK_SDHI0>;
-			resets = <&cpg R9A07G044L_CLK_SDHI0>;
+			clocks = <&cpg CPG_MOD R9A07G044_SDHI0_IMCLK>,
+				 <&cpg CPG_MOD R9A07G044_SDHI0_CLK_HS>,
+				 <&cpg CPG_MOD R9A07G044_SDHI0_IMCLK2>,
+				 <&cpg CPG_MOD R9A07G044_SDHI0_ACLK>;
+			clock-names = "core", "clkh", "cd", "aclk";
+			resets = <&cpg R9A07G044_SDHI0_IXRST>;
 			power-domains = <&cpg>;
 			status = "disabled";
 		};
 
 		sdhi1: sd@11c10000 {
 			compatible = "renesas,sdhi-r9a07g044l",
-				     "renesas,rcar-gen3-sdhi";
+					"renesas,sdhi-r9a07g044",
+				    "renesas,rcar-gen3-sdhi";
 			reg = <0x0 0x11c10000 0 0x10000>;
 			interrupts = <GIC_SPI 106 IRQ_TYPE_LEVEL_HIGH>,
 				     <GIC_SPI 107 IRQ_TYPE_LEVEL_HIGH>;
-			clocks = <&cpg CPG_MOD R9A07G044L_CLK_SDHI1>;
-			resets = <&cpg R9A07G044L_CLK_SDHI1>;
+			clocks = <&cpg CPG_MOD R9A07G044_SDHI1_IMCLK>,
+				 <&cpg CPG_MOD R9A07G044_SDHI1_CLK_HS>,
+				 <&cpg CPG_MOD R9A07G044_SDHI1_IMCLK2>,
+				 <&cpg CPG_MOD R9A07G044_SDHI1_ACLK>;
+			clock-names = "core", "clkh", "cd", "aclk";
+			resets = <&cpg R9A07G044_SDHI1_IXRST>;
 			power-domains = <&cpg>;
 			status = "disabled";
 		};
@@ -216,9 +469,9 @@
 			#address-cells = <1>;
 			#size-cells = <0>;
 			compatible = "renesas,riic-r9a07g044l",
-					"renesas,riic-rz";
-			reg = <0 0x10058000 0 0x44>;
-			interrupts = <GIC_SPI 350  IRQ_TYPE_LEVEL_HIGH>,
+					"renesas,riic-r9a07g044", "renesas,riic-rz";
+			reg = <0 0x10058000 0 0x400>;
+			interrupts = <GIC_SPI 350 IRQ_TYPE_LEVEL_HIGH>,
 				     <GIC_SPI 348 IRQ_TYPE_EDGE_RISING>,
 				     <GIC_SPI 349 IRQ_TYPE_EDGE_RISING>,
 				     <GIC_SPI 352 IRQ_TYPE_LEVEL_HIGH>,
@@ -226,9 +479,12 @@
 				     <GIC_SPI 351 IRQ_TYPE_LEVEL_HIGH>,
 				     <GIC_SPI 354 IRQ_TYPE_LEVEL_HIGH>,
 				     <GIC_SPI 355 IRQ_TYPE_LEVEL_HIGH>;
-			clocks = <&cpg CPG_MOD R9A07G044L_CLK_I2C0>;
+			interrupt-names = "tei", "ri", "ti", "spi", "sti",
+					  "naki", "ali", "tmoi";
+			clocks = <&cpg CPG_MOD R9A07G044_I2C0_PCLK>;
 			clock-frequency = <100000>;
-			resets = <&cpg R9A07G044L_CLK_I2C0>;
+			resets = <&cpg R9A07G044_I2C0_MRST>;
+			power-domains = <&cpg>;
 			status = "disabled";
 		};
 
@@ -236,9 +492,9 @@
 			#address-cells = <1>;
 			#size-cells = <0>;
 			compatible = "renesas,riic-r9a07g044l",
-					"renesas,riic-rz";
-			reg = <0 0x10058400 0 0x44>;
-			interrupts = <GIC_SPI 358  IRQ_TYPE_LEVEL_HIGH>,
+					"renesas,riic-r9a07g044", "renesas,riic-rz";
+			reg = <0 0x10058400 0 0x400>;
+			interrupts = <GIC_SPI 358 IRQ_TYPE_LEVEL_HIGH>,
 				     <GIC_SPI 356 IRQ_TYPE_EDGE_RISING>,
 				     <GIC_SPI 357 IRQ_TYPE_EDGE_RISING>,
 				     <GIC_SPI 360 IRQ_TYPE_LEVEL_HIGH>,
@@ -246,9 +502,12 @@
 				     <GIC_SPI 359 IRQ_TYPE_LEVEL_HIGH>,
 				     <GIC_SPI 362 IRQ_TYPE_LEVEL_HIGH>,
 				     <GIC_SPI 363 IRQ_TYPE_LEVEL_HIGH>;
-			clocks = <&cpg CPG_MOD R9A07G044L_CLK_I2C1>;
+			interrupt-names = "tei", "ri", "ti", "spi", "sti",
+					  "naki", "ali", "tmoi";
+			clocks = <&cpg CPG_MOD R9A07G044_I2C1_PCLK>;
 			clock-frequency = <100000>;
-			resets = <&cpg R9A07G044L_CLK_I2C1>;
+			resets = <&cpg R9A07G044_I2C1_MRST>;
+			power-domains = <&cpg>;
 			status = "disabled";
 		};
 
@@ -256,9 +515,9 @@
 			#address-cells = <1>;
 			#size-cells = <0>;
 			compatible = "renesas,riic-r9a07g044l",
-					"renesas,riic-rz";
-			reg = <0 0x10058800 0 0x44>;
-			interrupts = <GIC_SPI 366  IRQ_TYPE_LEVEL_HIGH>,
+					"renesas,riic-r9a07g044", "renesas,riic-rz";
+			reg = <0 0x10058800 0 0x400>;
+			interrupts = <GIC_SPI 366 IRQ_TYPE_LEVEL_HIGH>,
 				     <GIC_SPI 364 IRQ_TYPE_EDGE_RISING>,
 				     <GIC_SPI 365 IRQ_TYPE_EDGE_RISING>,
 				     <GIC_SPI 368 IRQ_TYPE_LEVEL_HIGH>,
@@ -266,9 +525,12 @@
 				     <GIC_SPI 367 IRQ_TYPE_LEVEL_HIGH>,
 				     <GIC_SPI 370 IRQ_TYPE_LEVEL_HIGH>,
 				     <GIC_SPI 371 IRQ_TYPE_LEVEL_HIGH>;
-			clocks = <&cpg CPG_MOD R9A07G044L_CLK_I2C2>;
+			interrupt-names = "tei", "ri", "ti", "spi", "sti",
+					  "naki", "ali", "tmoi";
+			clocks = <&cpg CPG_MOD R9A07G044_I2C2_PCLK>;
 			clock-frequency = <100000>;
-			resets = <&cpg R9A07G044L_CLK_I2C2>;
+			resets = <&cpg R9A07G044_I2C2_MRST>;
+			power-domains = <&cpg>;
 			status = "disabled";
 		};
 
@@ -276,9 +538,9 @@
 			#address-cells = <1>;
 			#size-cells = <0>;
 			compatible = "renesas,riic-r9a07g044l",
-					"renesas,riic-rz";
-			reg = <0 0x10058c00 0 0x44>;
-			interrupts = <GIC_SPI 374  IRQ_TYPE_LEVEL_HIGH>,
+					"renesas,riic-r9a07g044", "renesas,riic-rz";
+			reg = <0 0x10058c00 0 0x400>;
+			interrupts = <GIC_SPI 374 IRQ_TYPE_LEVEL_HIGH>,
 				     <GIC_SPI 372 IRQ_TYPE_EDGE_RISING>,
 				     <GIC_SPI 373 IRQ_TYPE_EDGE_RISING>,
 				     <GIC_SPI 376 IRQ_TYPE_LEVEL_HIGH>,
@@ -286,35 +548,71 @@
 				     <GIC_SPI 375 IRQ_TYPE_LEVEL_HIGH>,
 				     <GIC_SPI 378 IRQ_TYPE_LEVEL_HIGH>,
 				     <GIC_SPI 379 IRQ_TYPE_LEVEL_HIGH>;
-			clocks = <&cpg CPG_MOD R9A07G044L_CLK_I2C3>;
+			interrupt-names = "tei", "ri", "ti", "spi", "sti",
+					  "naki", "ali", "tmoi";
+			clocks = <&cpg CPG_MOD R9A07G044_I2C3_PCLK>;
 			clock-frequency = <100000>;
-			resets = <&cpg R9A07G044L_CLK_I2C3>;
+			resets = <&cpg R9A07G044_I2C3_MRST>;
+			power-domains = <&cpg>;
 			status = "disabled";
 		};
 
 		adc: adc@10059000 {
-			compatible = "renesas,adc-r9a07g044l";
+			compatible = "renesas,adc-r9a07g044l",
+						"renesas,r9a07g044-adc", "renesas,rzg2l-adc";
 			reg = <0 0x10059000 0 0x400>;
-			interrupts = <GIC_SPI 347  IRQ_TYPE_EDGE_RISING>;
-			clocks = <&cpg CPG_MOD R9A07G044L_CLK_ADC>;
-			resets = <&cpg R9A07G044L_CLK_ADC>;
+			interrupts = <GIC_SPI 347 IRQ_TYPE_EDGE_RISING>;
+			clocks = <&cpg CPG_MOD R9A07G044_ADC_ADCLK>,
+				 <&cpg CPG_MOD R9A07G044_ADC_PCLK>;
+			clock-names = "adclk", "pclk";
+			resets = <&cpg R9A07G044_ADC_PRESETN>,
+				 <&cpg R9A07G044_ADC_ADRST_N>;
+			reset-names = "presetn", "adrst-n";
 			power-domains = <&cpg>;
 			status = "disabled";
+
+			#address-cells = <1>;
+			#size-cells = <0>;
+
+			channel@0 {
+				reg = <0>;
+			};
+			channel@1 {
+				reg = <1>;
+			};
+			channel@2 {
+				reg = <2>;
+			};
+			channel@3 {
+				reg = <3>;
+			};
+			channel@4 {
+				reg = <4>;
+			};
+			channel@5 {
+				reg = <5>;
+			};
+			channel@6 {
+				reg = <6>;
+			};
+			channel@7 {
+				reg = <7>;
+			};
 		};
 
 		tsu: serial@10059400 {
-			compatible = "renesas,tsu-r9a07g044l";
+			compatible = "renesas,tsu-r9a07g044l",
+						"renesas,r9a07g044-tsu",
+				     "renesas,rzg2l-tsu";
 			reg = <0 0x10059400 0 0x400>;
-			clocks = <&cpg CPG_MOD R9A07G044L_CLK_TSU_PCLK>;
-			resets = <&cpg R9A07G044L_CLK_TSU_PCLK>;
-			clock-names = "tsu";
-			thermal-cal = <1>;
-			#thermal-sensor-cells = <0>;
+			clocks = <&cpg CPG_MOD R9A07G044_TSU_PCLK>;
+			resets = <&cpg R9A07G044_TSU_PRESETN>;
 			power-domains = <&cpg>;
+			#thermal-sensor-cells = <1>;
 		};
 
-		dma0: dma-controller@11800000 {
-			compatible = "renesas,rza-dma";
+		dmac: dma-controller@11820000 {
+			compatible = "renesas,rza-dma","renesas,r9a07g044-dmac","renesas,rz-dmac";
 			reg =   <0 0x11820000 0 0x10000>,
 				<0 0x11830000 0 0x10000>;
 
@@ -335,84 +633,119 @@
 				<GIC_SPI 139 IRQ_TYPE_EDGE_RISING>,
 				<GIC_SPI 140 IRQ_TYPE_EDGE_RISING>,
 				<GIC_SPI 141 IRQ_TYPE_EDGE_RISING>;
-			interrupt-names = "ch0", "ch1", "ch2", "ch3",
-				"ch4", "ch5", "ch6", "ch7",
-				"ch8", "ch9", "ch10", "ch11",
-				"ch12", "ch13", "ch14", "ch15",
-				"error";
+			interrupt-names = "error",
+					  "ch0", "ch1", "ch2", "ch3",
+					  "ch4", "ch5", "ch6", "ch7",
+					  "ch8", "ch9", "ch10", "ch11",
+					  "ch12", "ch13", "ch14", "ch15";
+			clocks = <&cpg CPG_MOD R9A07G044_DMAC_ACLK>,
+				 <&cpg CPG_MOD R9A07G044_DMAC_PCLK>;
+			clock-names = "main", "register";
+			power-domains = <&cpg>;
+			resets = <&cpg R9A07G044_DMAC_ARESETN>,
+				 <&cpg R9A07G044_DMAC_RST_ASYNC>;
+			reset-names = "arst", "rst_async";
 			dma-channels = <16>;
 			#dma-cells = <1>;
-			clocks = <&cpg CPG_MOD R9A07G044L_CLK_DMAC>;
-			resets = <&cpg R9A07G044L_CLK_DMAC>;
 		};
 
 		can: can@10050000 {
 			compatible = "renesas,rcar-gen3-canfd",
-					"renesas,r9a07g044l-canfd";
-			reg = <0 0x10050000 0 0x6000>;
-			interrupts =    <GIC_SPI 426 IRQ_TYPE_LEVEL_HIGH>,
-					<GIC_SPI 422 IRQ_TYPE_LEVEL_HIGH>,
-					<GIC_SPI 428 IRQ_TYPE_LEVEL_HIGH>,
-					<GIC_SPI 423 IRQ_TYPE_LEVEL_HIGH>,
-					<GIC_SPI 429 IRQ_TYPE_LEVEL_HIGH>,
-					<GIC_SPI 424 IRQ_TYPE_LEVEL_HIGH>,
-					<GIC_SPI 425 IRQ_TYPE_LEVEL_HIGH>,
-					<GIC_SPI 427 IRQ_TYPE_LEVEL_HIGH>;
-
-			clocks = <&cpg CPG_MOD R9A07G044L_CLK_CANFD>,
-				 <&cpg CPG_CORE R9A07G044L_CLK_P0>;
-			clock-names = "fck","canfd";
-			resets = <&cpg R9A07G044L_CLK_CANFD>;
-			power-dmains = <&cpg>;
+					"renesas,r9a07g044l-canfd",
+					"renesas,r9a07g044-canfd", "renesas,rzg2l-canfd";
+			reg = <0 0x10050000 0 0x8000>;
+			interrupts = <GIC_SPI 426 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 427 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 422 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 424 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 428 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 423 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 425 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 429 IRQ_TYPE_LEVEL_HIGH>;
+			interrupt-names = "g_err", "g_recc",
+					  "ch0_err", "ch0_rec", "ch0_trx",
+					  "ch1_err", "ch1_rec", "ch1_trx";
+			clocks = <&cpg CPG_MOD R9A07G044_CANFD_PCLK>,
+				 <&cpg CPG_CORE R9A07G044_CLK_P0_DIV2>,
+				 <&can_clk>;
+			clock-names = "fck", "canfd", "can_clk";
+			assigned-clocks = <&cpg CPG_CORE R9A07G044_CLK_P0_DIV2>;
+			assigned-clock-rates = <50000000>;
+			resets = <&cpg R9A07G044_CANFD_RSTP_N>,
+				 <&cpg R9A07G044_CANFD_RSTC_N>;
+			reset-names = "rstp_n", "rstc_n";
+			power-domains = <&cpg>;
 			status = "disabled";
+
 			channel0 {
-				renesas,channel = <0>;
 				status = "disabled";
 			};
 			channel1 {
-				renesas,channel = <1>;
 				status = "disabled";
 			};
 		};
 
 		eth0: ethernet@11c20000 {
-			compatible = "renesas,etheravb-r9a07g044l";
+			compatible = "renesas,etheravb-r9a07g044l",
+						"renesas,r9a07g044-gbeth",
+						"renesas,rzg2l-gbeth";
 			reg = <0 0x11c20000 0 0x10000>;
 			interrupts = <GIC_SPI 84 IRQ_TYPE_LEVEL_HIGH>,
 				     <GIC_SPI 85 IRQ_TYPE_LEVEL_HIGH>,
 				     <GIC_SPI 86 IRQ_TYPE_LEVEL_HIGH>;
-			clocks = <&cpg CPG_MOD R9A07G044L_CLK_ETH0>;
-			resets = <&cpg R9A07G044L_CLK_ETH0>;
-			power-domains = <&cpg>;
+			interrupt-names = "mux", "fil", "arp_ns";
 			phy-mode = "rgmii";
+			clocks = <&cpg CPG_MOD R9A07G044_ETH0_CLK_AXI>,
+				 <&cpg CPG_MOD R9A07G044_ETH0_CLK_CHI>,
+				 <&cpg CPG_CORE R9A07G044_CLK_HP>;
+			clock-names = "axi", "chi", "refclk";
+			resets = <&cpg R9A07G044_ETH0_RST_HW_N>;
+			power-domains = <&cpg>;
 			#address-cells = <1>;
 			#size-cells = <0>;
-			eth-pin-mode-switch = <&pinctrl 0>;
 			status = "disabled";
 		};
-		
+
 		eth1: ethernet@11c30000 {
-			compatible = "renesas,etheravb-r9a07g044l";
+			compatible = "renesas,etheravb-r9a07g044l",
+						"renesas,r9a07g044-gbeth",
+						"renesas,rzg2l-gbeth";
 			reg = <0 0x11c30000 0 0x10000>;
 			interrupts = <GIC_SPI 87 IRQ_TYPE_LEVEL_HIGH>,
 				     <GIC_SPI 88 IRQ_TYPE_LEVEL_HIGH>,
 				     <GIC_SPI 89 IRQ_TYPE_LEVEL_HIGH>;
-			clocks = <&cpg CPG_MOD R9A07G044L_CLK_ETH1>;
-			resets = <&cpg R9A07G044L_CLK_ETH1>;
-			power-domains = <&cpg>;
+			interrupt-names = "mux", "fil", "arp_ns";
 			phy-mode = "rgmii";
+			clocks = <&cpg CPG_MOD R9A07G044_ETH1_CLK_AXI>,
+				 <&cpg CPG_MOD R9A07G044_ETH1_CLK_CHI>,
+				 <&cpg CPG_CORE R9A07G044_CLK_HP>;
+			clock-names = "axi", "chi", "refclk";
+			resets = <&cpg R9A07G044_ETH1_RST_HW_N>;
+			power-domains = <&cpg>;
 			#address-cells = <1>;
 			#size-cells = <0>;
-			eth-pin-mode-switch = <&pinctrl 1>;
 			status = "disabled";
 		};
-
+		phyrst: usbphy-ctrl@11c40000 {
+			compatible = "renesas,r9a07g044-usbphy-ctrl",
+				     "renesas,rzg2l-usbphy-ctrl";
+			reg = <0 0x11c40000 0 0x10000>;
+			clocks = <&cpg CPG_MOD R9A07G044_USB_PCLK>;
+			resets = <&cpg R9A07G044_USB_PRESETN>;
+			power-domains = <&cpg>;
+			#reset-cells = <1>;
+			status = "disabled";
+		};
 		ohci0: usb@11c50000 {
 			compatible = "generic-ohci";
 			reg = <0 0x11c50000 0 0x100>;
 			interrupts = <GIC_SPI 91 IRQ_TYPE_LEVEL_HIGH>;
-			clocks = <&cpg CPG_MOD R9A07G044L_CLK_USB0>;
-			resets = <&cpg R9A07G044L_CLK_USB0>;
+			clocks = <&cpg CPG_MOD R9A07G044_USB_PCLK>,
+				 <&cpg CPG_MOD R9A07G044_USB_U2H0_HCLK>;
+			resets = <&phyrst 0>,
+				 <&cpg R9A07G044_USB_U2H0_HRESETN>;
+			phys = <&usb2_phy0 1>;
+			phy-names = "usb";
 			power-domains = <&cpg>;
 			status = "disabled";
 		};
@@ -421,18 +754,27 @@
 			compatible = "generic-ehci";
 			reg = <0 0x11c50100 0 0x100>;
 			interrupts = <GIC_SPI 92 IRQ_TYPE_LEVEL_HIGH>;
-			clocks = <&cpg CPG_MOD R9A07G044L_CLK_USB0>;
-			resets = <&cpg R9A07G044L_CLK_USB0>;
+			clocks = <&cpg CPG_MOD R9A07G044_USB_PCLK>,
+				 <&cpg CPG_MOD R9A07G044_USB_U2H0_HCLK>;
+			resets = <&phyrst 0>,
+				 <&cpg R9A07G044_USB_U2H0_HRESETN>;
+			phys = <&usb2_phy0 2>;
+			phy-names = "usb";
+			companion = <&ohci0>;
 			power-domains = <&cpg>;
 			status = "disabled";
 		};
 
-		ohci1: usb@e11c70000 {
+		ohci1: usb@11c70000 {
 			compatible = "generic-ohci";
 			reg = <0 0x11c70000 0 0x100>;
 			interrupts = <GIC_SPI 96 IRQ_TYPE_LEVEL_HIGH>;
-			clocks = <&cpg CPG_MOD R9A07G044L_CLK_USB1>;
-			resets = <&cpg R9A07G044L_CLK_USB1>;
+			clocks = <&cpg CPG_MOD R9A07G044_USB_PCLK>,
+				 <&cpg CPG_MOD R9A07G044_USB_U2H1_HCLK>;
+			resets = <&phyrst 1>,
+				 <&cpg R9A07G044_USB_U2H1_HRESETN>;
+			phys = <&usb2_phy1 1>;
+			phy-names = "usb";
 			power-domains = <&cpg>;
 			status = "disabled";
 		};
@@ -441,54 +783,75 @@
 			compatible = "generic-ehci";
 			reg = <0 0x11c70100 0 0x100>;
 			interrupts = <GIC_SPI 97 IRQ_TYPE_LEVEL_HIGH>;
-			clocks = <&cpg CPG_MOD R9A07G044L_CLK_USB1>;
-			resets = <&cpg R9A07G044L_CLK_USB1>;
+			clocks = <&cpg CPG_MOD R9A07G044_USB_PCLK>,
+				 <&cpg CPG_MOD R9A07G044_USB_U2H1_HCLK>;
+			resets = <&phyrst 1>,
+				 <&cpg R9A07G044_USB_U2H1_HRESETN>;
+			phys = <&usb2_phy1 2>;
+			phy-names = "usb";
+			companion = <&ohci1>;
 			power-domains = <&cpg>;
 			status = "disabled";
 		};
 
 		hsusb: usb@11c60000 {
-			compatible = "renesas,usbhs-r9a07g044l",
-				     "renesas,g2l-usbhs";
-			reg = <0 0x11c60000 0 0x724>;
-			interrupts = <GIC_SPI 100 IRQ_TYPE_EDGE_RISING>;
-			clocks = <&cpg CPG_MOD R9A07G044L_CLK_USB0>;
-			resets = <&cpg R9A07G044L_CLK_USB0>;
+			compatible = "renesas,usbhs-r9a07g044l","renesas,usbhs-r9a07g044",
+				     "renesas,rza2-usbhs";
+			reg = <0 0x11c60000 0 0x10000>;
+			interrupts = <GIC_SPI 100 IRQ_TYPE_EDGE_RISING>,
+				     <GIC_SPI 101 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 102 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 103 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&cpg CPG_MOD R9A07G044_USB_PCLK>,
+				 <&cpg CPG_MOD R9A07G044_USB_U2P_EXR_CPUCLK>;
+			resets = <&phyrst 0>,
+				 <&cpg R9A07G044_USB_U2P_EXL_SYSRST>;
 			renesas,buswait = <7>;
+			phys = <&usb2_phy0 3>;
+			phy-names = "usb";
 			power-domains = <&cpg>;
 			status = "disabled";
 		};
 
 		usb2_phy0: usb-phy@11c50200 {
-			compatible = "renesas,usb2-phy-r9a07g044l",
-				     "renesas,rcar-gen3-usb2-phy";
+			compatible = "renesas,usb2-phy-r9a07g044l","renesas,usb2-phy-r9a07g044",
+				    "renesas,rcar-gen3-usb2-phy","renesas,rzg2l-usb2-phy";
 			reg = <0 0x11c50200 0 0x700>;
-			interrupts = <GIC_SPI 90 IRQ_TYPE_LEVEL_HIGH>;
-			clocks = <&cpg CPG_MOD R9A07G044L_CLK_USB0>;
+			interrupts = <GIC_SPI 94 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&cpg CPG_MOD R9A07G044_USB_PCLK>,
+				 <&cpg CPG_MOD R9A07G044_USB_U2H0_HCLK>;
+			resets = <&phyrst 0>;
+			#phy-cells = <1>;
 			power-domains = <&cpg>;
-			resets = <&cpg R9A07G044L_CLK_USB0>;
-			#phy-cells = <0>;
 			status = "disabled";
 		};
 
-		usb2_phy1: usb-phy@11C70200 {
-			compatible = "renesas,usb2-phy-r9a07g044l",
-				     "renesas,rcar-gen3-usb2-phy";
-			reg = <0 0x11C70200 0 0x700>;
-			interrupts = <GIC_SPI 95 IRQ_TYPE_LEVEL_HIGH>;
-			clocks = <&cpg CPG_MOD R9A07G044L_CLK_USB1>;
+		usb2_phy1: usb-phy@11c70200 {
+			compatible = "renesas,usb2-phy-r9a07g044l","renesas,usb2-phy-r9a07g044",
+				     "renesas,rcar-gen3-usb2-phy","renesas,rzg2l-usb2-phy";
+			reg = <0 0x11c70200 0 0x700>;
+			interrupts = <GIC_SPI 99 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&cpg CPG_MOD R9A07G044_USB_PCLK>,
+				 <&cpg CPG_MOD R9A07G044_USB_U2H1_HCLK>;
+			resets = <&phyrst 1>;
+			#phy-cells = <1>;
 			power-domains = <&cpg>;
-			resets = <&cpg R9A07G044L_CLK_USB1>;
-			#phy-cells = <0>;
 			status = "disabled";
 		};
 
+
 		spibsc: spi@10060000 {
-			compatible = "renesas,r9a07g044l-spibsc";
-			reg = <0 0x10060000 0 0x78>, <0 0x20000000 0 0x10000000>;
+			compatible = "renesas,r9a07g044l-spibsc",
+					"renesas,r9a07g044-rpc-if",
+				    "renesas,rzg2l-rpc-if";
+			reg = <0 0x10060000 0 0x10000>,
+			      <0 0x20000000 0 0x10000000>,
+			      <0 0x10070000 0 0x10000>;
+			reg-names = "regs", "dirmap", "wbuf";
 			interrupts = <GIC_SPI 41 IRQ_TYPE_LEVEL_HIGH>;
-			clocks = <&cpg CPG_MOD R9A07G044L_CLK_SPI>;
-			resets = <&cpg R9A07G044L_CLK_SPI>;
+			clocks = <&cpg CPG_MOD R9A07G044_SPI_CLK2>,
+				 <&cpg CPG_MOD R9A07G044_SPI_CLK>;
+			resets = <&cpg R9A07G044_SPI_RST>;
 			power-domains = <&cpg>;
 			#address-cells = <1>;
 			#size-cells = <0>;
@@ -496,106 +859,159 @@
 		};
 
 		rspi0: spi@1004ac00 {
-			#address-cells = <1>;
-			#size-cells = <0>;
-			compatible = "renesas,rspi-r9a07g044l", "renesas,rspi-rz";
-			reg = <0 0x1004ac00 0 0x24>;
+			compatible = "renesas,rspi-r9a07g044l", "renesas,rspi-rz",
+						"renesas,r9a07g044-rspi";
+			reg = <0 0x1004ac00 0 0x400>;
 			interrupts = <GIC_SPI 415 IRQ_TYPE_LEVEL_HIGH>,
-			             <GIC_SPI 413 IRQ_TYPE_LEVEL_HIGH>,
-			             <GIC_SPI 414 IRQ_TYPE_LEVEL_HIGH>;
+				     <GIC_SPI 413 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 414 IRQ_TYPE_LEVEL_HIGH>;
 			interrupt-names = "error", "rx", "tx";
-			clocks = <&cpg CPG_MOD R9A07G044L_CLK_RSPI0>;
-			resets = <&cpg R9A07G044L_CLK_RSPI0>;
+			clocks = <&cpg CPG_MOD R9A07G044_RSPI0_CLKB>;
+			resets = <&cpg R9A07G044_RSPI0_RST>;
+			dmas = <&dmac 0x2e95>, <&dmac 0x2e96>;
+			dma-names = "tx", "rx";
 			power-domains = <&cpg>;
 			num-cs = <1>;
+			#address-cells = <1>;
+			#size-cells = <0>;
 			status = "disabled";
 		};
 
 		rspi1: spi@1004b000 {
-			#address-cells = <1>;
-			#size-cells = <0>;
-			compatible = "renesas,rspi-r9a07g044l", "renesas,rspi-rz";
-			reg = <0 0x1004b000 0 0x24>;
+			compatible = "renesas,rspi-r9a07g044l", "renesas,rspi-rz",
+						"renesas,r9a07g044-rspi";
+			reg = <0 0x1004b000 0 0x400>;
 			interrupts = <GIC_SPI 418 IRQ_TYPE_LEVEL_HIGH>,
-			             <GIC_SPI 416 IRQ_TYPE_LEVEL_HIGH>,
-			             <GIC_SPI 417 IRQ_TYPE_LEVEL_HIGH>;
+				     <GIC_SPI 416 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 417 IRQ_TYPE_LEVEL_HIGH>;
 			interrupt-names = "error", "rx", "tx";
-			clocks = <&cpg CPG_MOD R9A07G044L_CLK_RSPI1>;
-			resets = <&cpg R9A07G044L_CLK_RSPI1>;
+			clocks = <&cpg CPG_MOD R9A07G044_RSPI1_CLKB>;
+			resets = <&cpg R9A07G044_RSPI1_RST>;
+			dmas = <&dmac 0x2e99>, <&dmac 0x2e9a>;
+			dma-names = "tx", "rx";
 			power-domains = <&cpg>;
 			num-cs = <1>;
+			#address-cells = <1>;
+			#size-cells = <0>;
 			status = "disabled";
 		};
 
 		rspi2: spi@1004b400 {
-			#address-cells = <1>;
-			#size-cells = <0>;
-			compatible = "renesas,rspi-r9a07g044l", "renesas,rspi-rz";
-			reg = <0 0x1004b400 0 0x24>;
+			compatible = "renesas,rspi-r9a07g044l", "renesas,rspi-rz",
+						"renesas,r9a07g044-rspi";
+			reg = <0 0x1004b400 0 0x400>;
 			interrupts = <GIC_SPI 421 IRQ_TYPE_LEVEL_HIGH>,
-			             <GIC_SPI 419 IRQ_TYPE_LEVEL_HIGH>,
-			             <GIC_SPI 420 IRQ_TYPE_LEVEL_HIGH>;
+				     <GIC_SPI 419 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 420 IRQ_TYPE_LEVEL_HIGH>;
 			interrupt-names = "error", "rx", "tx";
-			clocks = <&cpg CPG_MOD R9A07G044L_CLK_RSPI2>;
-			resets = <&cpg R9A07G044L_CLK_RSPI2>;
+			clocks = <&cpg CPG_MOD R9A07G044_RSPI2_CLKB>;
+			resets = <&cpg R9A07G044_RSPI2_RST>;
+			dmas = <&dmac 0x2e9d>, <&dmac 0x2e9e>;
+			dma-names = "tx", "rx";
 			power-domains = <&cpg>;
 			num-cs = <1>;
+			#address-cells = <1>;
+			#size-cells = <0>;
 			status = "disabled";
 		};
 
-		gpt3: gpt@10048300 {
-			compatible = "renesas,gpt-r9a07g044l";
-			reg = <0 0x10048300 0 0xa4>;
-			#pwm-cells = <2>;
-			interrupts = <GIC_SPI 257 IRQ_TYPE_EDGE_RISING>,
-				<GIC_SPI 258 IRQ_TYPE_EDGE_RISING>,
-				<GIC_SPI 259 IRQ_TYPE_EDGE_RISING>,
-				<GIC_SPI 260 IRQ_TYPE_EDGE_RISING>,
-				<GIC_SPI 261 IRQ_TYPE_EDGE_RISING>,
-				<GIC_SPI 262 IRQ_TYPE_EDGE_RISING>,
-				<GIC_SPI 263 IRQ_TYPE_EDGE_RISING>,
-				<GIC_SPI 264 IRQ_TYPE_EDGE_RISING>,
-				<GIC_SPI 265 IRQ_TYPE_EDGE_RISING>,
-				<GIC_SPI 266 IRQ_TYPE_EDGE_RISING>,
-				<GIC_SPI 267 IRQ_TYPE_EDGE_RISING>,
-				<GIC_SPI 268 IRQ_TYPE_EDGE_RISING>,
-				<GIC_SPI 269 IRQ_TYPE_EDGE_RISING>;
-			interrupt-names = "gtcia", "gtcib", "gtcic", "gtcid",
-				"gtcie", "gtcif", "gtciada", "gtciadb",
-				"gtciv", "gtciu", "gtcih", "gtcil",
-				"gtdei";
-			clocks = <&cpg CPG_MOD R9A07G044L_CLK_GPT>;
-			resets =  <&cpg R9A07G044L_CLK_GPT>;
+
+		ssi0: ssi@10049c00 {
+			compatible = "renesas,rzg2l-ssi",
+						"renesas,r9a07g044-ssi",
+						"renesas,rz-ssi";
+			reg = <0 0x10049c00 0 0x400>;
+			interrupts = <GIC_SPI 326 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 327 IRQ_TYPE_EDGE_RISING>,
+				     <GIC_SPI 328 IRQ_TYPE_EDGE_RISING>;
+			interrupt-names = "int_req", "dma_rx", "dma_tx";
+			clocks = <&cpg CPG_MOD R9A07G044_SSI0_PCLK2>,
+				 <&cpg CPG_MOD R9A07G044_SSI0_PCLK_SFR>,
+				 <&audio_clk1>, <&audio_clk2>;
+			clock-names = "ssi", "ssi_sfr", "audio_clk1", "audio_clk2";
+			resets = <&cpg R9A07G044_SSI0_RST_M2_REG>;
+			dmas = <&dmac 0x2655>, <&dmac 0x2656>;
+			dma-names = "tx", "rx";
+			power-domains = <&cpg>;
+			#sound-dai-cells = <0>;
 			status = "disabled";
 		};
 
-		vspd0: vsp@10870000 {
-			compatible = "renesas,vsp2";
-			reg = <0 0x10870000 0 0x10000>;
-			interrupts = <GIC_SPI 149 IRQ_TYPE_LEVEL_HIGH>;
-			clocks = <&cpg CPG_MOD R9A07G044L_CLK_LCDC>;
-			resets = <&cpg R9A07G044L_CLK_LCDC>;
+		ssi1: ssi@1004A000 {
+			compatible = "renesas,rzg2l-ssi",
+						"renesas,r9a07g044-ssi",
+						"renesas,rz-ssi";
+			reg = <0 0x1004a000 0 0x400>;
+			interrupts = <GIC_SPI 330 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 331 IRQ_TYPE_EDGE_RISING>,
+				     <GIC_SPI 332 IRQ_TYPE_EDGE_RISING>;
+			interrupt-names = "int_req", "dma_rx", "dma_tx";
+			clocks = <&cpg CPG_MOD R9A07G044_SSI1_PCLK2>,
+				 <&cpg CPG_MOD R9A07G044_SSI1_PCLK_SFR>,
+				 <&audio_clk1>, <&audio_clk2>;
+			clock-names = "ssi", "ssi_sfr", "audio_clk1", "audio_clk2";
+			resets = <&cpg R9A07G044_SSI1_RST_M2_REG>;
+			dmas = <&dmac 0x2659>, <&dmac 0x265a>;
+			dma-names = "tx", "rx";
 			power-domains = <&cpg>;
-			renesas,fcp = <&fcpvd0>;
+			#sound-dai-cells = <0>;
+			status = "disabled";
 		};
 
-		fcpvd0: fcp@10880000 {
-			compatible = "renesas,fcpv";
-			reg = <0 0x10880000 0 0x10000>;
-			clocks = <&cpg CPG_MOD R9A07G044L_CLK_LCDC>;
-			resets = <&cpg R9A07G044L_CLK_LCDC>;
+		ssi2: ssi@1004A400 {
+			compatible = "renesas,rzg2l-ssi",
+						"renesas,r9a07g044-ssi",
+						"renesas,rz-ssi";
+			reg = <0 0x1004a400 0 0x400>;
+			interrupts = <GIC_SPI 334 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 337 IRQ_TYPE_EDGE_RISING>;
+			interrupt-names = "int_req", "dma_rt";
+			clocks = <&cpg CPG_MOD R9A07G044_SSI2_PCLK2>,
+				 <&cpg CPG_MOD R9A07G044_SSI2_PCLK_SFR>,
+				 <&audio_clk1>, <&audio_clk2>;
+			clock-names = "ssi", "ssi_sfr", "audio_clk1", "audio_clk2";
+			resets = <&cpg R9A07G044_SSI2_RST_M2_REG>;
+			dmas = <&dmac 0x265f>;
+			dma-names = "rt";
 			power-domains = <&cpg>;
+			#sound-dai-cells = <0>;
+			status = "disabled";
 		};
 
-		du: display@feb00000 {
-			compatible = "renesas,du-r9a04g077l";
-			reg = <0 0x10890000 0 0x10000>;
-			interrupts = <GIC_SPI 152 IRQ_TYPE_LEVEL_HIGH>;
-			clocks = <&cpg CPG_MOD R9A07G044L_CLK_LCDC>;
-			clock-names = "du.0";
-			resets = <&cpg R9A07G044L_CLK_LCDC>;
-			reset-names = "du.0";
-			vsps = <&vspd0 0>;
+		ssi3: ssi@1004A800 {
+			compatible = "renesas,rzg2l-ssi",
+						"renesas,r9a07g044-ssi",
+						"renesas,rz-ssi";
+			reg = <0 0x1004a800 0 0x400>;
+			interrupts = <GIC_SPI 338 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 339 IRQ_TYPE_EDGE_RISING>,
+				     <GIC_SPI 340 IRQ_TYPE_EDGE_RISING>;
+			interrupt-names = "int_req", "dma_rx", "dma_tx";
+			clocks = <&cpg CPG_MOD R9A07G044_SSI3_PCLK2>,
+				 <&cpg CPG_MOD R9A07G044_SSI3_PCLK_SFR>,
+				 <&audio_clk1>, <&audio_clk2>;
+			clock-names = "ssi", "ssi_sfr", "audio_clk1", "audio_clk2";
+			resets = <&cpg R9A07G044_SSI3_RST_M2_REG>;
+			dmas = <&dmac 0x2661>, <&dmac 0x2662>;
+			dma-names = "tx", "rx";
+			power-domains = <&cpg>;
+			#sound-dai-cells = <0>;
+			status = "disabled";
+		};
+		cru: video@10830000 {
+			compatible = "renesas,r9a07g044-cru", "renesas,rzg2l-cru";
+			reg = <0 0x10830000 0 0x400>;
+			clocks = <&cpg CPG_MOD R9A07G044_CRU_VCLK>,
+				 <&cpg CPG_MOD R9A07G044_CRU_PCLK>,
+				 <&cpg CPG_MOD R9A07G044_CRU_ACLK>;
+			clock-names = "video", "apb", "axi";
+			interrupts = <GIC_SPI 167 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 168 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 169 IRQ_TYPE_LEVEL_HIGH>;
+			interrupt-names = "image_conv", "image_conv_err", "axi_mst_err";
+			resets = <&cpg R9A07G044_CRU_PRESETN>,
+				 <&cpg R9A07G044_CRU_ARESETN>;
+			reset-names = "presetn", "aresetn";
 			power-domains = <&cpg>;
 			status = "disabled";
 
@@ -604,27 +1020,39 @@
 				#size-cells = <0>;
 
 				port@0 {
+					#address-cells = <1>;
+					#size-cells = <0>;
+
 					reg = <0>;
-					du_out_rgb: endpoint {
+					cruparallel: endpoint@0 {
+						reg = <0>;
 					};
 				};
 
 				port@1 {
+					#address-cells = <1>;
+					#size-cells = <0>;
+
 					reg = <1>;
-					du_out_dsi0: endpoint {
-						remote-endpoint = <&dsi0_in>;
+					crucsi2: endpoint@0 {
+						reg = <0>;
+						remote-endpoint = <&csi2cru>;
 					};
 				};
 			};
 		};
 
-		dsi0: dsi@10860000 {
-			compatible = "renesas,r9a07g044l-mipi-dsi";
-			reg =	<0 0x10860000 0 0x10000>, /* LINK */
-				<0 0x10850000 0 0x10000>; /* DPHY */
-
-			clocks = <&cpg CPG_MOD R9A07G044L_CLK_MIPI_DSI>;
-			resets = <&cpg R9A07G044L_CLK_MIPI_DSI>;
+		csi2: csi2@10830400 {
+			compatible = "renesas,r9a07g044-csi2", "renesas,rzg2l-csi2";
+			reg = <0 0x10830400 0 0xfc00>;
+			interrupts = <GIC_SPI 166 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&cpg CPG_MOD R9A07G044_CRU_SYSCLK>,
+				 <&cpg CPG_MOD R9A07G044_CRU_VCLK>,
+				 <&cpg CPG_MOD R9A07G044_CRU_PCLK>;
+			clock-names = "system", "video", "apb";
+			resets = <&cpg R9A07G044_CRU_PRESETN>,
+				 <&cpg R9A07G044_CRU_CMN_RSTB>;
+			reset-names = "presetn", "cmn-rstb";
 			power-domains = <&cpg>;
 			status = "disabled";
 
@@ -634,100 +1062,155 @@
 
 				port@0 {
 					reg = <0>;
-					dsi0_in: endpoint {
-					remote-endpoint = <&du_out_dsi0>;
-					};
 				};
 
 				port@1 {
+					#address-cells = <1>;
+					#size-cells = <0>;
 					reg = <1>;
-					dsi0_out: endpoint {
+
+					csi2cru: endpoint@0 {
+						reg = <0>;
+						remote-endpoint = <&crucsi2>;
 					};
 				};
 			};
 		};
 
-		ssi0: ssi@10049c00 {
-			#sound-dai-cells = <0>;
-			compatible = "renesas,rzg2l-ssi";
-			reg = <0 0x10049c00 0 0x00000028>;
-			interrupts = <GIC_SPI 326 IRQ_TYPE_LEVEL_HIGH>,
-				     <GIC_SPI 327 IRQ_TYPE_EDGE_RISING>,
-				     <GIC_SPI 328 IRQ_TYPE_EDGE_RISING>;
-			interrupt-names = "int", "rx", "tx";
-			clocks = <&cpg CPG_MOD R9A07G044L_CLK_SSI0> ,
-				 <&audio_clk1>,
-				 <&audio_clk2>;
-			clock-names = "ssi", "audio_clk1", "audio_clk2";
-			resets = <&cpg R9A07G044L_CLK_SSI0>;
+		sysc: system-controller@11020000 {
+			compatible = "renesas,r9a07g044-sysc";
+			reg = <0 0x11020000 0 0x10000>;
+			interrupts = <GIC_SPI 42 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 43 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 44 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 45 IRQ_TYPE_LEVEL_HIGH>;
+			interrupt-names = "lpm_int", "ca55stbydone_int",
+					  "cm33stbyr_int", "ca55_deny";
+			status = "disabled";
+		};
+
+		gpu: gpu@11840000 {
+			compatible = "renesas,r9a07g044-mali",
+				     "arm,mali-bifrost";
+			reg = <0x0 0x11840000 0x0 0x10000>;
+			interrupts = <GIC_SPI 154 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 155 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 153 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 156 IRQ_TYPE_LEVEL_HIGH>;
+			interrupt-names = "job", "mmu", "gpu", "event";
+			clocks = <&cpg CPG_MOD R9A07G044_GPU_CLK>,
+				 <&cpg CPG_MOD R9A07G044_GPU_AXI_CLK>,
+				 <&cpg CPG_MOD R9A07G044_GPU_ACE_CLK>;
+			clock-names = "gpu", "bus", "bus_ace";
+			power-domains = <&cpg>;
+			resets = <&cpg R9A07G044_GPU_RESETN>,
+				 <&cpg R9A07G044_GPU_AXI_RESETN>,
+				 <&cpg R9A07G044_GPU_ACE_RESETN>;
+			reset-names = "rst", "axi_rst", "ace_rst";
+			operating-points-v2 = <&gpu_opp_table>;
+		};
+
+		ostm0: timer@12801000 {
+			compatible = "renesas,r9a07g044-ostm",
+				     "renesas,ostm";
+			reg = <0x0 0x12801000 0x0 0x400>;
+			interrupts = <GIC_SPI 46 IRQ_TYPE_EDGE_RISING>;
+			clocks = <&cpg CPG_MOD R9A07G044_OSTM0_PCLK>;
+			resets = <&cpg R9A07G044_OSTM0_PRESETZ>;
+			power-domains = <&cpg>;
+			status = "okay";
+		};
+
+		ostm1: timer@12801400 {
+			compatible = "renesas,r9a07g044-ostm",
+				     "renesas,ostm";
+			reg = <0x0 0x12801400 0x0 0x400>;
+			interrupts = <GIC_SPI 47 IRQ_TYPE_EDGE_RISING>;
+			clocks = <&cpg CPG_MOD R9A07G044_OSTM1_PCLK>;
+			resets = <&cpg R9A07G044_OSTM1_PRESETZ>;
 			power-domains = <&cpg>;
+			status = "okay";
+		};
+
+		ostm2: timer@12801800 {
+			compatible = "renesas,r9a07g044-ostm",
+				     "renesas,ostm";
+			reg = <0x0 0x12801800 0x0 0x400>;
+			interrupts = <GIC_SPI 48 IRQ_TYPE_EDGE_RISING>;
+			clocks = <&cpg CPG_MOD R9A07G044_OSTM2_PCLK>;
+			resets = <&cpg R9A07G044_OSTM2_PRESETZ>;
+			power-domains = <&cpg>;
+			status = "okay";
+		};
+
+		poega: poega@10048800 {
+			compatible = "renesas,poeg-r9a07g044";
+			reg = <0 0x10048800 0 0x04>;
+			interrupts = <GIC_SPI 322 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&cpg CPG_MOD R9A07G044_POEG_A_CLKP>;
+			power-domains = <&cpg>;
+			resets =  <&cpg R9A07G044_POEG_A_RST>;
 			status = "disabled";
 		};
 
-		ssi1: ssi@1004A000 {
-			#sound-dai-cells = <0>;
-			compatible = "renesas,rzg2l-ssi";
-			reg = <0 0x1004A000 0 0x00000028>;
-			interrupts = <GIC_SPI 330 IRQ_TYPE_LEVEL_HIGH>,
-				     <GIC_SPI 331 IRQ_TYPE_EDGE_RISING>,
-				     <GIC_SPI 332 IRQ_TYPE_EDGE_RISING>;
-			interrupt-names = "int", "rx", "tx";
-			clocks = <&cpg CPG_MOD R9A07G044L_CLK_SSI1>,
-				 <&audio_clk1>,
-				 <&audio_clk2>;
-			clock-names = "ssi", "audio_clk1", "audio_clk2";
-			resets = <&cpg R9A07G044L_CLK_SSI1>;
+		poegb: poegb@10048C00 {
+			compatible = "renesas,poeg-r9a07g044";
+			reg = <0 0x10048C00 0 0x04>;
+			interrupts = <GIC_SPI 323 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&cpg CPG_MOD R9A07G044_POEG_B_CLKP>;
 			power-domains = <&cpg>;
+			resets =  <&cpg R9A07G044_POEG_B_RST>;
 			status = "disabled";
 		};
 
-		ssi2: ssi@1004A400 {
-			#sound-dai-cells = <0>;
-			compatible = "renesas,rzg2l-ssi";
-			reg = <0 0x1004A400 0 0x00000028>;
-			interrupts = <GIC_SPI 334 IRQ_TYPE_LEVEL_HIGH>,
-				     <GIC_SPI 335 IRQ_TYPE_EDGE_RISING>,
-				     <GIC_SPI 336 IRQ_TYPE_EDGE_RISING>;
-			interrupt-names = "int", "rx", "tx";
-			clocks = <&cpg CPG_MOD R9A07G044L_CLK_SSI2>,
-				 <&audio_clk1>,
-				 <&audio_clk2>;
-			clock-names = "ssi", "audio_clk1", "audio_clk2";
-			resets = <&cpg R9A07G044L_CLK_SSI2>;
+		poegc: poegc@10049000 {
+			compatible = "renesas,poeg-r9a07g044";
+			reg = <0 0x10049000 0 0x04>;
+			interrupts = <GIC_SPI 324 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&cpg CPG_MOD R9A07G044_POEG_C_CLKP>;
 			power-domains = <&cpg>;
+			resets =  <&cpg R9A07G044_POEG_C_RST>;
 			status = "disabled";
 		};
 
-		ssi3: ssi@1004A800 {
-			#sound-dai-cells = <0>;
-			compatible = "renesas,rzg2l-ssi";
-			reg = <0 0x1004A800 0 0x00000028>;
-			interrupts = <GIC_SPI 338 IRQ_TYPE_LEVEL_HIGH>,
-				     <GIC_SPI 339 IRQ_TYPE_EDGE_RISING>,
-				     <GIC_SPI 340 IRQ_TYPE_EDGE_RISING>;
-			interrupt-names = "int", "rx", "tx";
-			clocks = <&cpg CPG_MOD R9A07G044L_CLK_SSI3>,
-				 <&audio_clk1>,
-				 <&audio_clk2>;
-			clock-names = "ssi", "audio_clk1", "audio_clk2";
-			resets = <&cpg R9A07G044L_CLK_SSI3>;
+		poegd: poegd@10049400 {
+			compatible = "renesas,poeg-r9a07g044";
+			reg = <0 0x10049400 0 0x04>;
+			interrupts = <GIC_SPI 325 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&cpg CPG_MOD R9A07G044_POEG_D_CLKP>;
 			power-domains = <&cpg>;
+			resets =  <&cpg R9A07G044_POEG_D_RST>;
 			status = "disabled";
 		};
+
 	};
 
 	thermal-zones {
-		sensor_thermal: sensor-thermal {
+		cpu-thermal {
 			polling-delay-passive = <250>;
 			polling-delay = <1000>;
-			sustainable-power = <3874>;
 			thermal-sensors = <&tsu 0>;
+			sustainable-power = <717>;
+
+			cooling-maps {
+				map0 {
+					trip = <&target>;
+					cooling-device = <&cpu0 0 2>;
+					contribution = <1024>;
+				};
+			};
 
 			trips {
 				sensor_crit: sensor-crit {
-				temperature = <110000>;
-				hysteresis = <1000>;
-				type = "critical";
+					temperature = <125000>;
+					hysteresis = <1000>;
+					type = "critical";
+				};
+
+				target: trip-point {
+					temperature = <100000>;
+					hysteresis = <1000>;
+					type = "passive";
 				};
 			};
 		};
@@ -735,15 +1218,9 @@
 
 	timer {
 		compatible = "arm,armv8-timer";
-		interrupts-extended =
-		<&gic GIC_PPI 13
-			(GIC_CPU_MASK_SIMPLE(2) | IRQ_TYPE_LEVEL_LOW)>,
-		<&gic GIC_PPI 14
-			(GIC_CPU_MASK_SIMPLE(2) | IRQ_TYPE_LEVEL_LOW)>,
-		<&gic GIC_PPI 11
-			(GIC_CPU_MASK_SIMPLE(2) | IRQ_TYPE_LEVEL_LOW)>,
-		<&gic GIC_PPI 10
-			(GIC_CPU_MASK_SIMPLE(2) | IRQ_TYPE_LEVEL_LOW)>;
+		interrupts-extended = <&gic GIC_PPI 13 IRQ_TYPE_LEVEL_LOW>,
+				      <&gic GIC_PPI 14 IRQ_TYPE_LEVEL_LOW>,
+				      <&gic GIC_PPI 11 IRQ_TYPE_LEVEL_LOW>,
+				      <&gic GIC_PPI 10 IRQ_TYPE_LEVEL_LOW>;
 	};
 };
-
diff --git a/arch/arm/dts/smarc-rzg2l.dts b/arch/arm/dts/smarc-rzg2l.dts
index 7de4cc7b4a..404bc64a75 100644
--- a/arch/arm/dts/smarc-rzg2l.dts
+++ b/arch/arm/dts/smarc-rzg2l.dts
@@ -1,149 +1,18 @@
-// SPDX-License-Identifier: GPL-2.0
+// SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause)
 /*
- * Device Tree Source for the GR-Peach board
+ * Device Tree Source for the RZ/G2L SMARC EVK board
  *
- * Copyright (C) 2017 Jacopo Mondi <jacopo+renesas@jmondi.org>
- * Copyright (C) 2016 Renesas Electronics
+ * Copyright (C) 2023 Renesas Electronics Corp.
  */
 
 /dts-v1/;
-#include <dt-bindings/gpio/gpio.h>
-#include "r9a07g044l.dtsi"
-#include "r9a07g044l-u-boot.dtsi"
-#include <dt-bindings/pinctrl/rzg2l-pinctrl.h>
+#include "r9a07g044l2.dtsi"
+#include "rzg2l-smarc-som.dtsi"
+#include "rzg2l-smarc-pinfunction.dtsi"
+#include "rz-smarc-common.dtsi"
+#include "rzg2l-smarc.dtsi"
 
 / {
-	model = "smarc-rzg2l";
-	compatible = "renesas,r9a07g044l", "renesas,rzg2l";
-
-	aliases {
-		serial0 = &scif0;
-		spi0 = &spibsc;
-	};
-
-	chosen {
-		stdout-path = "serial0:115200n8";
-	};
-
-	memory@48000000 {
-		device_type = "memory";
-		/* first 128MB is reserved for secure area. */
-		reg = <0 0x48000000 0 0x78000000>;
-	};
-
-};
-
-&pinctrl {
-	eth0_pins: eth0 {
-		pinmux = <RZG2L_PINMUX(20, 0, 1)>,
-				<RZG2L_PINMUX(20, 1, 1)>,
-				<RZG2L_PINMUX(20, 2, 1)>,
-				<RZG2L_PINMUX(21, 0, 1)>,
-				<RZG2L_PINMUX(21, 1, 1)>,
-				<RZG2L_PINMUX(22, 0, 1)>,
-				<RZG2L_PINMUX(22, 1, 1)>,
-				<RZG2L_PINMUX(23, 0, 1)>,
-				<RZG2L_PINMUX(23, 1, 1)>,
-				<RZG2L_PINMUX(24, 0, 1)>,
-				<RZG2L_PINMUX(24, 1, 1)>,
-				<RZG2L_PINMUX(25, 0, 1)>,
-				<RZG2L_PINMUX(25, 1, 1)>,
-				<RZG2L_PINMUX(26, 0, 1)>,
-				<RZG2L_PINMUX(26, 1, 1)>,
-				<RZG2L_PINMUX(27, 0, 1)>,
-				<RZG2L_PINMUX(27, 1, 1)>,
-				<RZG2L_PINMUX(28, 0, 1)>,
-				<RZG2L_PINMUX(28, 1, 1)>;
-	};
-
-	i2c1_pins: i2c1 {
-		pinmux = <RZG2L_PINMUX(14, 2, 1)>,
-				<RZG2L_PINMUX(14, 3, 1)>;
-	};
-
-};
-
-&xinclk {
-        clock-frequency = <24000000>;
-};
-
-&scif0 {
-	status = "okay";
-	clock = <100000000>;
-};
-
-&sdhi0 {
-	/* pinctrl placeholder
-	 * If this channel is used for interfacing with a SD card, a power enable
-	 * pin (SD0_PWR_EN) must be defined.
-	 * The SD0_PWR_EN pin is associated with P4_1.
-	 * A HIGH signal on SD0_PWR_EN will enable supply voltage for SD card.
-	*/
-	bus-width = <8>;
-	status = "okay";
-};
-
-&sdhi1 {
-	/* pinctrl placeholder
-	 * A power enable pin (SD1_PWR_EN) must be defined to interface with a
-	 * SD card.
-	 * The SD1_PWR_EN pin is associated with P39_2.
-	 * A HIGH signal on SD1_PWR_EN will enable supply voltage for SD card.
-	*/
-	bus-width = <4>;
-	status = "okay";
-};
-
-&eth0 {
-	pinctrl-names = "default";
-	pinctrl-0 = <&eth0_pins>;
-	status = "okay";
-
-	phy-handle = <&phy0>;
-	phy-mode = "rgmii";
-	phy0: ethernet-phy@7 {
-		reg = <7>;
-	};
-};
-
-&i2c1 {
-	pinctrl-names = "default";
-	pinctrl-0 = <&i2c1_pins>;
-	status = "okay";
-
-	adv7535: hdmi@3d {
-		compatible = "adi,adv7535";
-		reg = <0x3d>;
-	};
-};
-
-&spibsc {
-	num-cs = <1>;
-	status = "okay";
-	spi-max-frequency = <40000000>;
-	#address-cells = <1>;
-	#size-cells = <0>;
-	flash0: spi-flash@0 {
-		#address-cells = <1>;
-		#size-cells = <1>;
-		compatible = "spi-flash", "jedec,spi-nor";
-		spi-max-frequency = <40000000>;
-		spi-tx-bus-width = <1>;
-		spi-rx-bus-width = <1>;
-		reg = <0>;
-		status = "okay";
-	};
-};
-
-&ehci0 {
-	status = "okay";
-};
-
-&ehci1 {
-	status = "okay";
-};
-
-&wdt0 {
-	status = "okay";
-	timeout-sec = <60>;
+	model = "Renesas SMARC EVK based on r9a07g044l2";
+	compatible = "renesas,smarc-evk", "renesas,r9a07g044l2", "renesas,r9a07g044";
 };
diff --git a/drivers/clk/renesas/clk-rzg2l.c b/drivers/clk/renesas/clk-rzg2l.c
index 7a0de87b03..a1b838c836 100644
--- a/drivers/clk/renesas/clk-rzg2l.c
+++ b/drivers/clk/renesas/clk-rzg2l.c
@@ -3,9 +3,9 @@
  * Renesas RZ/G2L CLK driver
  *
  * Copyright (C) 2020 Renesas Electronics Corp.
- * 
+ *
  * Based on the following driver from Linux kernel:
- * r8a774a1 Clock Pulse Generator / Module Standby and Software Reset 
+ * Clock Pulse Generator / Module Standby and Software Reset
  */
 
 #include <common.h>
@@ -45,19 +45,20 @@ static int rzg2l_clk_enable(struct clk *clk)
 	const struct mssr_mod_clk *clock;
 	uint32_t value=0;
 	int ret=0;
-	
+
 	if (!is_mod_clk(clk)) {
 		debug("CLK: not a module clock \n");
 		return -1;
 	}
-	
+
 	/* Set write_mask bit, set clk enable, then check monitor status */
 	mod_clk_get(clk, priv->info, &clock);
 	value = (MSSR_ON(clock->bit) << 16) | MSSR_ON(clock->bit);
 	setbits_le32(priv->base + CLK_ON_R(MSSR_OFF(clock->bit) * 4), value);
 	ret = wait_for_bit_le32(priv->base + CLK_MON_R(MSSR_OFF(clock->bit) * 4),
 					MSSR_ON(clock->bit), 1, 100, 0);
-	
+	/* reset the module*/
+	setbits_le32(priv->base + CLK_RST_R(MSSR_OFF(clock->bit) * 4), value);
 	return ret;
 }
 
@@ -72,7 +73,7 @@ static int rzg2l_clk_disable(struct clk *clk)
 		debug("CLK: not a module clock \n");
 		return -1;
 	}
-	
+
 	mod_clk_get(clk, priv->info, &clock);
 	value = MSSR_ON(clock->bit) << 16;
 	iowrite32(value, priv->base + 0x500 + MSSR_OFF(clock->bit) * 4);
@@ -84,7 +85,7 @@ static int rzg2l_clk_disable(struct clk *clk)
 
 static ulong rzg2l_clk_set_rate(struct clk *clk, ulong rate)
 {
-	
+
 	return 0;
 }
 
@@ -99,13 +100,13 @@ int rzg2l_clk_probe(struct udevice *dev)
 	struct rzg2l_clk_priv *priv = dev_get_priv(dev);
 	struct cpg_mssr_info *info =
 		(struct cpg_mssr_info *)dev_get_driver_data(dev);
-	
+
 	priv->info = info;
 	priv->base = dev_read_addr_ptr(dev);
 	if (!priv->base)
 		return -EINVAL;
 
-	ret = clk_get_by_name(dev, "xinclk", &priv->clk_extal);
+	ret = clk_get_by_name(dev, "extal", &priv->clk_extal);
 	if (ret < 0)
 		return ret;
 
diff --git a/drivers/clk/renesas/r9a07g044l-cpg-mssr.c b/drivers/clk/renesas/r9a07g044l-cpg-mssr.c
index 132fcc0673..3fe646e955 100644
--- a/drivers/clk/renesas/r9a07g044l-cpg-mssr.c
+++ b/drivers/clk/renesas/r9a07g044l-cpg-mssr.c
@@ -1,8 +1,7 @@
 /* SPDX-License-Identifier: GPL-2.0 */
-/*	
+/*
  * R-Car R9A07G044L Clock Module
-vers/clk/renesas/r9a07g044l-cpg-mssr.c *
- * Copyright (C) 2020 Renesas Electronics Corp.
+ * Copyright (C) 2023 Renesas Electronics Corp.
  *
  * This program is free software; you can redistribute it and/or modify
  * it under the terms of the GNU General Public License as published by
@@ -22,7 +21,7 @@ extern const struct clk_ops rzg2l_clk_ops;
 
 enum clk_ids {
 	/* Core Clock Outputs exported to DT */
-	LAST_DT_CORE_CLK = R9A07G044L_OSCCLK,
+	LAST_DT_CORE_CLK = R9A07G044_CLK_P0_DIV2,
 
 	/* External Input Clocks */
 	CLK_XINCLK,
@@ -150,11 +149,11 @@ static const char *sel_shdi[]	= {".clk800fix_c", ".clk533_c",
 				   ".clk800fix_cdiv2", ".clk533_cdiv2"};
 
 static const struct cpg_core_clk r9a07g044l_core_clks[] = {
-	/* External Clock Inputs */
+		/* External Clock Inputs */
 	DEF_INPUT("xinclk",	 CLK_XINCLK),
 
-	/* Internal Core Clocks */
-	DEF_FIXED(".osc", R9A07G044L_OSCCLK, CLK_XINCLK, 1, 1),
+		/* Internal Core Clocks */
+	DEF_FIXED(".osc", R9A07G044_OSCCLK, CLK_XINCLK, 1, 1),
 	DEF_FIXED(".osc_div1000", CLK_OSC_DIV1000, CLK_XINCLK, 1, 1000),
 	DEF_BASE(".pll1", CLK_PLL1, CLK_TYPE_PLL1, CLK_XINCLK),
 	DEF_BASE(".pll2", CLK_PLL2, CLK_TYPE_PLL2, CLK_XINCLK),
@@ -226,208 +225,231 @@ static const struct cpg_core_clk r9a07g044l_core_clks[] = {
 			sel_pll6_2, 2, 0),
 
 	/* Core output clk*/
-	DEF_DIV("I", R9A07G044L_CLK_I, CLK_SEL_PLL1,
+	DEF_DIV("I", R9A07G044_CLK_I, CLK_SEL_PLL1,
 			DIVPL1, NULL, CLK_DIVIDER_POWER_OF_TWO),
-	DEF_DIV("I2", R9A07G044L_CLK_I2, CLK200FIX_CD,
+	DEF_DIV("I2", R9A07G044_CLK_I2, CLK200FIX_CD,
 		DIVPL3CLK200FIX, dtable_3b, 0),
-	DEF_DIV("G", R9A07G044L_CLK_G, CLK_SEL_G2,
+	DEF_DIV("G", R9A07G044_CLK_G, CLK_SEL_G2,
 			DIVGPU, NULL, CLK_DIVIDER_POWER_OF_TWO),
-	DEF_FIXED("S0", R9A07G044L_CLK_S0, CLK_SEL_PLL4, 1, 2),
-	DEF_FIXED("S1", R9A07G044L_CLK_S0, CLK_SEL_PLL4, 1, 4),
-	DEF_FIXED("SPI0", R9A07G044L_CLK_SPI0, CLK533_CD, 1, 2),
-	DEF_FIXED("SPI1", R9A07G044L_CLK_SPI1, CLK533_CD, 1, 4),
-	DEF_MUX("SD0", R9A07G044L_CLK_SD0, SEL_SDHI0,
+	DEF_FIXED("S0", R9A07G044_CLK_S0, CLK_SEL_PLL4, 1, 2),
+	DEF_FIXED("S1", R9A07G044_CLK_S0, CLK_SEL_PLL4, 1, 4),
+	DEF_FIXED("SPI0", R9A07G044_CLK_SPI0, CLK533_CD, 1, 2),
+	DEF_FIXED("SPI1", R9A07G044_CLK_SPI1, CLK533_CD, 1, 4),
+	DEF_MUX("SD0", R9A07G044_CLK_SD0, SEL_SDHI0,
 			sel_shdi, 4, 0),
-	DEF_MUX("SD1", R9A07G044L_CLK_SD1, SEL_SDHI1,
+	DEF_MUX("SD1", R9A07G044_CLK_SD1, SEL_SDHI1,
 			sel_shdi, 4, 0),
-	DEF_FIXED("M0", R9A07G044L_CLK_M0, CLK200FIX_CD, 1, 1),
-	DEF_FIXED("M1", R9A07G044L_CLK_M1, CLK_SEL_PLL5_1, 1, 1),
-	DEF_FIXED("M2", R9A07G044L_CLK_M2, CLK533FIX_CD, 1, 2),
-	DEF_2DIV("M3", R9A07G044L_CLK_M3, CLK_SEL_PLL5_4,
+	DEF_FIXED("M0", R9A07G044_CLK_M0, CLK200FIX_CD, 1, 1),
+	DEF_FIXED("M1", R9A07G044_CLK_M1, CLK_SEL_PLL5_1, 1, 1),
+	DEF_FIXED("M2", R9A07G044_CLK_M2, CLK533FIX_CD, 1, 2),
+	DEF_2DIV("M3", R9A07G044_CLK_M3, CLK_SEL_PLL5_4,
 		DIVDSIA, DIVDSIB, dtable_2b, dtable_4b, 0),
-	DEF_DIV("M4", R9A07G044L_CLK_M4, CLK533FIX_LPCLK,
+	DEF_DIV("M4", R9A07G044_CLK_M4, CLK533FIX_LPCLK,
 			DIVDSILPCL, divdsilpcl, 0),
-	DEF_MUX("HP", R9A07G044L_CLK_HP, SEL_ETH, sel_eth, 2, 0),
-	DEF_FIXED("TSU", R9A07G044L_CLK_TSU, CLK800FIX_C, 1, 10),
-	DEF_FIXED("ZT", R9A07G044L_CLK_ZT, CLK100FIX_CD, 1, 1),
-	DEF_DIV("P0", R9A07G044L_CLK_P0, CLK100FIX_C,
+	DEF_MUX("HP", R9A07G044_CLK_HP, SEL_ETH, sel_eth, 2, 0),
+	DEF_FIXED("TSU", R9A07G044_CLK_TSU, CLK800FIX_C, 1, 10),
+	DEF_FIXED("ZT", R9A07G044_CLK_ZT, CLK100FIX_CD, 1, 1),
+	DEF_DIV("P0", R9A07G044_CLK_P0, CLK100FIX_C,
 			DIVPL2A, dtable_3b, 0),
-	DEF_DIV("P1", R9A07G044L_CLK_P1, CLK200FIX_CD,
+	DEF_DIV("P1", R9A07G044_CLK_P1, CLK200FIX_CD,
 			DIVPL3B, dtable_3b, 0),
-	DEF_DIV("P2", R9A07G044L_CLK_P2, CLK100FIX_CD,
+	DEF_DIV("P2", R9A07G044_CLK_P2, CLK100FIX_CD,
 			DIVPL3A, dtable_3b, 0),
-	DEF_FIXED("AT", R9A07G044L_CLK_AT, CLK800FIX_CD, 1, 2),
+	DEF_FIXED("AT", R9A07G044_CLK_AT, CLK800FIX_CD, 1, 2),
 };
 
 static struct mssr_mod_clk r9a07g044l_mod_clks[] = {
-	DEF_MOD("gic",		R9A07G044L_CLK_GIC600,
-				R9A07G044L_CLK_P1,
+	DEF_MOD("gic",		R9A07G044_GIC600_GICCLK,
+				R9A07G044_CLK_P1,
 				MSSR(5, BIT(0), (BIT(0) | BIT(1)))),
-	DEF_MOD("ia55",		R9A07G044L_CLK_IA55,
-				R9A07G044L_CLK_P1,
+	DEF_MOD("ia55",		R9A07G044_IA55_PCLK,
+				R9A07G044_CLK_P1,
 				MSSR(6, (BIT(0) | BIT(1)), BIT(0))),
-	DEF_MOD("syc",		R9A07G044L_CLK_SYC,
-				CLK_XINCLK,
-				MSSR(10, BIT(0), BIT(0))),
-	DEF_MOD("dmac",		R9A07G044L_CLK_DMAC,
-				R9A07G044L_CLK_P1,
+	DEF_MOD("ia55",		R9A07G044_IA55_CLK,
+				R9A07G044_CLK_P1,
+				MSSR(6, (BIT(0) | BIT(1)), BIT(0))),
+
+	DEF_MOD("dmac_aclk",R9A07G044_DMAC_ACLK,
+				R9A07G044_CLK_P1,
 				MSSR(11, (BIT(0) | BIT(1)),
 					 (BIT(0) | BIT(1)))),
-	DEF_MOD("sysc",		R9A07G044L_CLK_SYSC,
-				CLK_XINCLK,
-				MSSR(12, (BIT(0) | BIT(1)),
-					 (BIT(0) | BIT(1) | BIT(2)))),
-	DEF_MOD("mtu3",		R9A07G044L_CLK_MTU,
-				R9A07G044L_CLK_P0,
+	DEF_MOD("dmac_aclk",R9A07G044_DMAC_PCLK,
+				R9A07G044_CLK_P1,
+				MSSR(11, (BIT(0) | BIT(1)),
+					 (BIT(0) | BIT(1)))),
+
+	DEF_MOD("mtu3",		R9A07G044_MTU_X_MCK_MTU3,
+				R9A07G044_CLK_P0,
 				MSSR(14, BIT(0), BIT(0))),
-	DEF_MOD("gpt",		R9A07G044L_CLK_GPT,
-				R9A07G044L_CLK_P0,
+	DEF_MOD("gpt",		R9A07G044_GPT_PCLK,
+				R9A07G044_CLK_P0,
 				MSSR(16, BIT(0), BIT(0))),
-	DEF_MOD("eth0",		R9A07G044L_CLK_ETH0,
-				R9A07G044L_CLK_HP,
+	DEF_MOD("eth0",		R9A07G044_ETH0_CLK_AXI,
+				R9A07G044_CLK_HP,
 				MSSR(31, BIT(0), BIT(0))),
-	DEF_MOD("eth1",		R9A07G044L_CLK_ETH1,
-				R9A07G044L_CLK_HP,
+	DEF_MOD("eth1",		R9A07G044_ETH1_CLK_AXI,
+				R9A07G044_CLK_HP,
 				MSSR(31, BIT(1), BIT(1))),
-	DEF_MOD("i2c0",		R9A07G044L_CLK_I2C0,
-				R9A07G044L_CLK_P0,
+	DEF_MOD("i2c0",		R9A07G044_I2C0_PCLK,
+				R9A07G044_CLK_P0,
 				MSSR(32, BIT(0), BIT(0))),
-	DEF_MOD("i2c1",		R9A07G044L_CLK_I2C1,
-				R9A07G044L_CLK_P0,
+	DEF_MOD("i2c1",		R9A07G044_I2C1_PCLK,
+				R9A07G044_CLK_P0,
 				MSSR(32, BIT(1), BIT(1))),
-	DEF_MOD("i2c2",		R9A07G044L_CLK_I2C2,
-				R9A07G044L_CLK_P0,
+	DEF_MOD("i2c2",		R9A07G044_I2C2_PCLK,
+				R9A07G044_CLK_P0,
 				MSSR(32, BIT(2), BIT(2))),
-	DEF_MOD("i2c3",		R9A07G044L_CLK_I2C3,
-				R9A07G044L_CLK_P0,
+	DEF_MOD("i2c3",		R9A07G044_I2C3_PCLK,
+				R9A07G044_CLK_P0,
 				MSSR(32, BIT(3), BIT(3))),
-	DEF_MOD("scif0",	R9A07G044L_CLK_SCIF0,
-				R9A07G044L_CLK_P0,
+	DEF_MOD("scif0",	R9A07G044_SCIF0_CLK_PCK,
+				R9A07G044_CLK_P0,
 				MSSR(33, BIT(0), BIT(0))),
-	DEF_MOD("scif1",	R9A07G044L_CLK_SCIF1,
-				R9A07G044L_CLK_P0,
+	DEF_MOD("scif1",	R9A07G044_SCIF1_CLK_PCK,
+				R9A07G044_CLK_P0,
 				MSSR(33, BIT(1), BIT(1))),
-	DEF_MOD("scif2",	R9A07G044L_CLK_SCIF2,
-				R9A07G044L_CLK_P0,
+	DEF_MOD("scif2",	R9A07G044_SCIF2_CLK_PCK,
+				R9A07G044_CLK_P0,
 				MSSR(33, BIT(2), BIT(2))),
-	DEF_MOD("scif3",	R9A07G044L_CLK_SCIF3,
-				R9A07G044L_CLK_P0,
+	DEF_MOD("scif3",	R9A07G044_SCIF3_CLK_PCK,
+				R9A07G044_CLK_P0,
 				MSSR(33, BIT(3), BIT(3))),
-	DEF_MOD("scif4",	R9A07G044L_CLK_SCIF4,
-				R9A07G044L_CLK_P0,
+	DEF_MOD("scif4",	R9A07G044_SCIF4_CLK_PCK,
+				R9A07G044_CLK_P0,
 				MSSR(33, BIT(4), BIT(4))),
-	DEF_MOD("sci0",		R9A07G044L_CLK_SCI0,
-				R9A07G044L_CLK_P0,
+	DEF_MOD("sci0",		R9A07G044_SCI0_CLKP,
+				R9A07G044_CLK_P0,
 				MSSR(34, BIT(0), BIT(0))),
-	DEF_MOD("sci1",		R9A07G044L_CLK_SCI1,
-				R9A07G044L_CLK_P0,
+	DEF_MOD("sci1",		R9A07G044_SCI1_CLKP,
+				R9A07G044_CLK_P0,
 				MSSR(34, BIT(1), BIT(1))),
-	DEF_MOD("gpio",		R9A07G044L_CLK_GPIO,
+	DEF_MOD("gpio",		R9A07G044_GPIO_HCLK,
 				CLK_XINCLK,
 				MSSR(38, BIT(0),
 					(BIT(0) | BIT(1) | BIT(2)))),
-	DEF_MOD("sdhi0",	R9A07G044L_CLK_SDHI0,
-				R9A07G044L_CLK_SD0,
-				MSSR(21, (BIT(0) | BIT(1) | BIT(2) | BIT(3)),
+	DEF_MOD("sdhi0_imclk",	R9A07G044_SDHI0_IMCLK,
+				R9A07G044_CLK_SD0,
+				MSSR(21, (BIT(0)),
+					 BIT(0))),
+	DEF_MOD("sdhi0_imclk2",	R9A07G044_SDHI0_IMCLK2,
+				R9A07G044_CLK_SD0,
+				MSSR(21, (BIT(1)),
+					 BIT(0))),
+	DEF_MOD("sdhi0_clk_hs",	R9A07G044_SDHI0_CLK_HS,
+				R9A07G044_CLK_SD0,
+				MSSR(21, (BIT(2)),
+					 BIT(0))),
+	DEF_MOD("sdhi0_aclk",	R9A07G044_SDHI0_ACLK,
+				R9A07G044_CLK_SD0,
+				MSSR(21, (BIT(3)),
+					 BIT(0))),
+	DEF_MOD("sdhi1_imclk",	R9A07G044_SDHI1_IMCLK,
+				R9A07G044_CLK_SD1,
+				MSSR(21, (BIT(4)),
+					 BIT(1))),
+	DEF_MOD("sdhi1_imclk2",	R9A07G044_SDHI1_IMCLK2,
+				R9A07G044_CLK_SD1,
+				MSSR(21, (BIT(5)),
+					 BIT(1))),
+	DEF_MOD("sdhi1_clk_hs",	R9A07G044_SDHI1_CLK_HS,
+				R9A07G044_CLK_SD1,
+				MSSR(21, (BIT(6)),
+					 BIT(1))),
+	DEF_MOD("sdhi1_aclk",	R9A07G044_SDHI1_ACLK,
+				R9A07G044_CLK_SD1,
+				MSSR(21, (BIT(7)),
+					 BIT(1))),
+
+	DEF_MOD("usb0_host",R9A07G044_USB_U2H0_HCLK,
+				R9A07G044_CLK_P1,
+				MSSR(30, BIT(0),
 					 BIT(0))),
-	DEF_MOD("sdhi1",	R9A07G044L_CLK_SDHI1,
-				R9A07G044L_CLK_SD1,
-				MSSR(21, (BIT(4) | BIT(5) | BIT(6) | BIT(7)),
+	DEF_MOD("usb1_host",R9A07G044_USB_U2H1_HCLK,
+				R9A07G044_CLK_P1,
+				MSSR(30, BIT(1),
 					 BIT(1))),
-	DEF_MOD("usb0",		R9A07G044L_CLK_USB0,
-				R9A07G044L_CLK_P1,
-				MSSR(30, (BIT(0) | BIT(2) | BIT(3)),
-					 (BIT(0) | BIT(2) | BIT(3)))),
-	DEF_MOD("usb1",		R9A07G044L_CLK_USB1,
-				R9A07G044L_CLK_P1,
-				MSSR(30, (BIT(1) | BIT(3)),
-					 (BIT(1) | BIT(3)))),
-	DEF_MOD("canfd",	R9A07G044L_CLK_CANFD,
-				R9A07G044L_CLK_P0,
+	DEF_MOD("usb0_func",R9A07G044_USB_U2P_EXR_CPUCLK,
+				R9A07G044_CLK_P1,
+				MSSR(30, BIT(2),
+					 BIT(2))),
+	DEF_MOD("usb_pclk",R9A07G044_USB_PCLK,
+				R9A07G044_CLK_P1,
+				MSSR(30, BIT(3),
+					 BIT(3))),
+
+
+
+	DEF_MOD("canfd",	R9A07G044_CANFD_PCLK,
+				R9A07G044_CLK_P0,
 				MSSR(37, BIT(0), (BIT(0) | BIT(1)))),
-	DEF_MOD("ssi0",		R9A07G044L_CLK_SSI0,
-				R9A07G044L_CLK_P0,
+	DEF_MOD("ssi0_pclk",		R9A07G044_SSI0_PCLK2,
+				R9A07G044_CLK_P0,
 				MSSR(28, (BIT(0) | BIT(1)), BIT(0))),
-	DEF_MOD("ssi1",		R9A07G044L_CLK_SSI1,
-				R9A07G044L_CLK_P0,
+	DEF_MOD("ssi0_sfr",		R9A07G044_SSI0_PCLK_SFR,
+				R9A07G044_CLK_P0,
+				MSSR(28, (BIT(0) | BIT(1)), BIT(0))),
+	DEF_MOD("ssi1_pclk",		R9A07G044_SSI1_PCLK2,
+				R9A07G044_CLK_P0,
+				MSSR(28, (BIT(2) | BIT(3)), BIT(1))),
+	DEF_MOD("ssi1_sfr",		R9A07G044_SSI1_PCLK_SFR,
+				R9A07G044_CLK_P0,
 				MSSR(28, (BIT(2) | BIT(3)), BIT(1))),
-	DEF_MOD("ssi2",		R9A07G044L_CLK_SSI2,
-				R9A07G044L_CLK_P0,
+	DEF_MOD("ssi2_pclk",		R9A07G044_SSI2_PCLK2,
+				R9A07G044_CLK_P0,
 				MSSR(28, (BIT(4) | BIT(5)), BIT(2))),
-	DEF_MOD("ssi3",		R9A07G044L_CLK_SSI3,
-				R9A07G044L_CLK_P0,
+	DEF_MOD("ssi2_sfr",		R9A07G044_SSI2_PCLK_SFR,
+				R9A07G044_CLK_P0,
+				MSSR(28, (BIT(4) | BIT(5)), BIT(2))),
+	DEF_MOD("ssi3_pclk",		R9A07G044_SSI3_PCLK2,
+				R9A07G044_CLK_P0,
+				MSSR(28, (BIT(6) | BIT(7)), BIT(3))),
+	DEF_MOD("ssi3_sfr",		R9A07G044_SSI3_PCLK_SFR,
+				R9A07G044_CLK_P0,
 				MSSR(28, (BIT(6) | BIT(7)), BIT(3))),
-	DEF_MOD("mhu",		R9A07G044L_CLK_MHU,
-				R9A07G044L_CLK_P1,
-				MSSR(8, BIT(0), BIT(0))),
-	DEF_MOD("ostm0",	R9A07G044L_CLK_OSTM0,
-				R9A07G044L_CLK_P0,
+	DEF_MOD("ostm0",	R9A07G044_OSTM0_PCLK,
+				R9A07G044_CLK_P0,
 				MSSR(13, BIT(0), BIT(0))),
-	DEF_MOD("ostm1",	R9A07G044L_CLK_OSTM1,
-				R9A07G044L_CLK_P0,
+	DEF_MOD("ostm1",	R9A07G044_OSTM1_PCLK,
+				R9A07G044_CLK_P0,
 				MSSR(13, BIT(1), BIT(1))),
-	DEF_MOD("ostm2",	R9A07G044L_CLK_OSTM2,
-				R9A07G044L_CLK_P0,
+	DEF_MOD("ostm2",	R9A07G044_OSTM2_PCLK,
+				R9A07G044_CLK_P0,
 				MSSR(13, BIT(2), BIT(3))),
-	DEF_MOD("wdt0",		R9A07G044L_CLK_WDT0,
-				R9A07G044L_CLK_P0,
-				MSSR(18, (BIT(0) | BIT(1)), BIT(0))),
-	DEF_MOD("wdt1",		R9A07G044L_CLK_WDT1,
-				R9A07G044L_CLK_P0,
-				MSSR(18, (BIT(2) | BIT(3)), BIT(1))),
-	DEF_MOD("wdt2",		R9A07G044L_CLK_WDT2,
-				R9A07G044L_CLK_P0,
-				MSSR(18, (BIT(4) | BIT(5)), BIT(2))),
-	DEF_MOD("wdt_pon",	R9A07G044L_CLK_WDT_PON,
-				R9A07G044L_CLK_P0,
-				MSSR(18, (BIT(6) | BIT(7)), BIT(3))),
-	DEF_MOD("gpu",		R9A07G044L_CLK_GPU,
-				R9A07G044L_CLK_G,
+	DEF_MOD("wdt0_pclk", R9A07G044_WDT0_PCLK,
+				R9A07G044_CLK_P0,
+				MSSR(18, (BIT(0)), BIT(0))),
+	DEF_MOD("wdt0_clk", R9A07G044_WDT0_CLK,
+				R9A07G044_CLK_P0,
+				MSSR(18, (BIT(1)), BIT(0))),
+	DEF_MOD("wdt1_pclk", R9A07G044_WDT1_PCLK,
+				R9A07G044_CLK_P0,
+				MSSR(18, (BIT(2)), BIT(1))),
+	DEF_MOD("wdt1_clk", R9A07G044_WDT1_CLK,
+				R9A07G044_CLK_P0,
+				MSSR(18, (BIT(3)), BIT(1))),
+	DEF_MOD("wdt2_pclk", R9A07G044_WDT2_PCLK,
+				R9A07G044_CLK_P0,
+				MSSR(18, (BIT(4)), BIT(2))),
+	DEF_MOD("wdt2_clk", R9A07G044_WDT2_CLK,
+				R9A07G044_CLK_P0,
+				MSSR(18, (BIT(5)), BIT(2))),
+	DEF_MOD("gpu_clk",		R9A07G044_GPU_CLK,
+				R9A07G044_CLK_G,
 				MSSR(22, (BIT(0) | BIT(1) | BIT(2)),
 					 (BIT(0) | BIT(1) | BIT(2)))),
-	DEF_MOD("isu",		R9A07G044L_CLK_ISU,
-				R9A07G044L_CLK_P1,
-				MSSR(23, (BIT(0) | BIT(1)), (BIT(0) | BIT(1)))),
-	DEF_MOD("h264",		R9A07G044L_CLK_H264,
-				R9A07G044L_CLK_M0,
-				MSSR(24, BIT(0), (BIT(0) | BIT(1)))),
-	DEF_MOD("cru",		R9A07G044L_CLK_CRU,
-				R9A07G044L_CLK_M2,
-				MSSR(25, (BIT(0) | BIT(1) | BIT(3) | BIT(4)),
+	DEF_MOD("gpu_axi_clk",		R9A07G044_GPU_AXI_CLK,
+				R9A07G044_CLK_G,
+				MSSR(22, (BIT(0) | BIT(1) | BIT(2)),
 					 (BIT(0) | BIT(1) | BIT(2)))),
-	DEF_MOD("mipi_dsi",	R9A07G044L_CLK_MIPI_DSI,
-				R9A07G044L_CLK_M3,
-				MSSR(26, (BIT(0) | BIT(1) | BIT(2) | BIT(3)
-							  | BIT(4) | BIT(5)),
+	DEF_MOD("gpu_cce_clk",		R9A07G044_GPU_ACE_CLK,
+				R9A07G044_CLK_G,
+				MSSR(22, (BIT(0) | BIT(1) | BIT(2)),
 					 (BIT(0) | BIT(1) | BIT(2)))),
-	DEF_MOD("lcdc",		R9A07G044L_CLK_LCDC,
-				R9A07G044L_CLK_M3,
-				MSSR(27, (BIT(0) | BIT(1)), BIT(0))),
-	DEF_MOD("src",		R9A07G044L_CLK_SRC,
-				R9A07G044L_CLK_P0,
-				MSSR(29, BIT(0), BIT(0))),
-	DEF_MOD("rspi0",	R9A07G044L_CLK_RSPI0,
-				R9A07G044L_CLK_P0,
-				MSSR(36, BIT(0), BIT(0))),
-	DEF_MOD("rspi1",	R9A07G044L_CLK_RSPI1,
-				R9A07G044L_CLK_P0,
-				MSSR(36, BIT(1), BIT(1))),
-	DEF_MOD("rspi2",	R9A07G044L_CLK_RSPI2,
-				R9A07G044L_CLK_P0,
-				MSSR(36, BIT(2), BIT(2))),
-	DEF_MOD("adc",		R9A07G044L_CLK_ADC,
-				R9A07G044L_CLK_TSU,
-				MSSR(42, (BIT(0) | BIT(1)), (BIT(0) | BIT(1)))),
-	DEF_MOD("tsu",		R9A07G044L_CLK_TSU_PCLK,
-				R9A07G044L_CLK_TSU,
-				MSSR(43, BIT(0), BIT(0))),
-	DEF_MOD("spi-multi",	R9A07G044L_CLK_SPI,
-				R9A07G044L_CLK_SPI1,
-				MSSR(21, (BIT(0) | BIT(1)), BIT(0))),
+
 };
 
 static const unsigned int r9a07g044l_crit_mod_clks[] = {
-	CLK_MODE_BASE + R9A07G044L_CLK_GIC600,
+	CLK_MODE_BASE + R9A07G044_GIC600_GICCLK,
 };
 
 /* clock type, register offset1, register offset2, register offset3*/
diff --git a/drivers/net/ravb.c b/drivers/net/ravb.c
index 5a5a242d51..ac54c05a27 100644
--- a/drivers/net/ravb.c
+++ b/drivers/net/ravb.c
@@ -744,6 +744,7 @@ static const struct udevice_id ravb_ids[] = {
 	{ .compatible = "renesas,etheravb-r9a07g054l" },
 	{ .compatible = "renesas,etheravb-r9a07g043u" },
 	{ .compatible = "renesas,etheravb-r9a07g043f" },
+	{ .compatible = "renesas,r9a07g044-gbeth" },
 	{ }
 };
 
diff --git a/drivers/phy/phy-rcar-gen3.c b/drivers/phy/phy-rcar-gen3.c
index 8c59631428..5018509c07 100644
--- a/drivers/phy/phy-rcar-gen3.c
+++ b/drivers/phy/phy-rcar-gen3.c
@@ -148,6 +148,7 @@ static int rcar_gen3_phy_remove(struct udevice *dev)
 
 static const struct udevice_id rcar_gen3_phy_of_match[] = {
 	{ .compatible = "renesas,rcar-gen3-usb2-phy", },
+	{ .compatible = "renesas,usb2-phy-r9a07g044", },
 	{ },
 };
 
diff --git a/drivers/pinctrl/renesas/pfc-rzg2l.c b/drivers/pinctrl/renesas/pfc-rzg2l.c
index f295394750..129377a073 100644
--- a/drivers/pinctrl/renesas/pfc-rzg2l.c
+++ b/drivers/pinctrl/renesas/pfc-rzg2l.c
@@ -2,7 +2,7 @@
 /*
  * Pin controller driver for Renesas RZ/G2L SoCs.
  *
- * Copyright (C) 2020 Renesas Electronics Corporation
+ * Copyright (C) 2023 Renesas Electronics Corporation
  */
 
 #include <common.h>
@@ -77,7 +77,7 @@ static int rzg2l_pinctrl_set_state(struct udevice *dev, struct udevice *config)
 
 	for (i = 0 ; i < count; i++) {
 		cells[i] = fdt32_to_cpu(data[i]);
-		func = (cells[i] >> 12) & 0xf;
+		func = (cells[i] >> 16) & 0xf;
 		port = (cells[i] / RZG2L_MAX_PINS_PER_PORT) & 0x1ff;
 		pin = cells[i] % RZG2L_MAX_PINS_PER_PORT;
 		if (func > 5 || port >= port_max || pin >= RZG2L_MAX_PINS_PER_PORT) {
diff --git a/drivers/power/regulator/regulator_common.c b/drivers/power/regulator/regulator_common.c
index 93d8196b38..15d243d7b5 100644
--- a/drivers/power/regulator/regulator_common.c
+++ b/drivers/power/regulator/regulator_common.c
@@ -1,6 +1,6 @@
 // SPDX-License-Identifier: GPL-2.0+
 /*
- * Copyright (C) 2019 Disruptive Technologies Research AS
+ * Copyright (C) 2023 Disruptive Technologies Research AS
  * Sven Schwermer <sven.svenschwermer@disruptive-technologies.com>
  */
 
@@ -66,6 +66,7 @@ int regulator_common_set_enable(const struct udevice *dev,
 	debug("%s: dev='%s', enable=%d, delay=%d, has_gpio=%d\n", __func__,
 	      dev->name, enable, dev_pdata->startup_delay_us,
 	      dm_gpio_is_valid(&dev_pdata->gpio));
+#if CONFIG_RENESAS_REGULATOR_GPIO_ENABLE
 	/* Enable GPIO is optional */
 	if (!dm_gpio_is_valid(&dev_pdata->gpio)) {
 		if (!enable)
@@ -79,7 +80,7 @@ int regulator_common_set_enable(const struct udevice *dev,
 		      enable);
 		return ret;
 	}
-
+#endif
 	if (enable && dev_pdata->startup_delay_us)
 		udelay(dev_pdata->startup_delay_us);
 	debug("%s: done\n", __func__);
diff --git a/drivers/serial/serial_sh.c b/drivers/serial/serial_sh.c
index 2fc661920b..3feec3a00c 100644
--- a/drivers/serial/serial_sh.c
+++ b/drivers/serial/serial_sh.c
@@ -215,6 +215,7 @@ static const struct udevice_id sh_serial_id[] ={
 	{.compatible = "renesas,sci", .data = PORT_SCI},
 	{.compatible = "renesas,scif", .data = PORT_SCIF},
 	{.compatible = "renesas,scifa", .data = PORT_SCIFA},
+	{.compatible = "renesas,scif-r9a07g044", .data = PORT_SCIF},
 	{}
 };
 
diff --git a/drivers/watchdog/rzg2l_wdt.c b/drivers/watchdog/rzg2l_wdt.c
index b6a7b75699..586af1342b 100644
--- a/drivers/watchdog/rzg2l_wdt.c
+++ b/drivers/watchdog/rzg2l_wdt.c
@@ -62,6 +62,7 @@ static const struct udevice_id rzg2l_wdt_match[] = {
 	{ .compatible = "renesas,r9a07g054l-wdt", },
 	{ .compatible = "renesas,r9a07g043u-wdt", },
 	{ .compatible = "renesas,r9a07g043f-wdt", },
+	{ .compatible = "renesas,r9a07g044-wdt", },
 	{ /* sentinel */ }
 };
 
diff --git a/include/dt-bindings/clock/r9a07g044l-cpg.h b/include/dt-bindings/clock/r9a07g044l-cpg.h
index 679d4c1e0a..088fdea565 100644
--- a/include/dt-bindings/clock/r9a07g044l-cpg.h
+++ b/include/dt-bindings/clock/r9a07g044l-cpg.h
@@ -1,88 +1,220 @@
-/* SPDX-License-Identifier: GPL-2.0
+/* SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause)
  *
- * Copyright (C) 2020 Renesas Electronics Corp.
+ * Copyright (C) 2023 Renesas Electronics Corp.
  */
-#ifndef __DT_BINDINGS_CLOCK_R9A07G044L_CPG_MSSR_H__
-#define __DT_BINDINGS_CLOCK_R9A07G044L_CPG_MSSR_H__
+#ifndef __DT_BINDINGS_CLOCK_R9A07G044_CPG_H__
+#define __DT_BINDINGS_CLOCK_R9A07G044_CPG_H__
 
 #include <dt-bindings/clock/renesas-cpg-mssr.h>
 
-/* r9a07g044l CPG Core Clocks */
-#define R9A07G044L_CLK_I		0
-#define R9A07G044L_CLK_I2		1
-#define R9A07G044L_CLK_G		2
-#define R9A07G044L_CLK_S0		3
-#define R9A07G044L_CLK_S1		4
-#define R9A07G044L_CLK_SPI0		5
-#define R9A07G044L_CLK_SPI1		6
-#define R9A07G044L_CLK_SD0		7
-#define R9A07G044L_CLK_SD1		8
-#define R9A07G044L_CLK_M0		9
-#define R9A07G044L_CLK_M1		10
-#define R9A07G044L_CLK_M2		11
-#define R9A07G044L_CLK_M3		12
-#define R9A07G044L_CLK_M4		13
-#define R9A07G044L_CLK_HP		14
-#define R9A07G044L_CLK_TSU		15
-#define R9A07G044L_CLK_ZT		16
-#define R9A07G044L_CLK_P0		17
-#define R9A07G044L_CLK_P1		18
-#define R9A07G044L_CLK_P2		19
-#define R9A07G044L_CLK_AT		20
-#define R9A07G044L_OSCCLK		21
+/* R9A07G044 CPG Core Clocks */
+#define R9A07G044_CLK_I			0
+#define R9A07G044_CLK_I2		1
+#define R9A07G044_CLK_G			2
+#define R9A07G044_CLK_S0		3
+#define R9A07G044_CLK_S1		4
+#define R9A07G044_CLK_SPI0		5
+#define R9A07G044_CLK_SPI1		6
+#define R9A07G044_CLK_SD0		7
+#define R9A07G044_CLK_SD1		8
+#define R9A07G044_CLK_M0		9
+#define R9A07G044_CLK_M1		10
+#define R9A07G044_CLK_M2		11
+#define R9A07G044_CLK_M3		12
+#define R9A07G044_CLK_M4		13
+#define R9A07G044_CLK_HP		14
+#define R9A07G044_CLK_TSU		15
+#define R9A07G044_CLK_ZT		16
+#define R9A07G044_CLK_P0		17
+#define R9A07G044_CLK_P1		18
+#define R9A07G044_CLK_P2		19
+#define R9A07G044_CLK_AT		20
+#define R9A07G044_OSCCLK		21
+#define R9A07G044_CLK_P0_DIV2		22
 
-/* r9a07g044l Module Clocks */
+/* R9A07G044 Module Clocks */
+#define R9A07G044_CA55_SCLK		0
+#define R9A07G044_CA55_PCLK		1
+#define R9A07G044_CA55_ATCLK		2
+#define R9A07G044_CA55_GICCLK		3
+#define R9A07G044_CA55_PERICLK		4
+#define R9A07G044_CA55_ACLK		5
+#define R9A07G044_CA55_TSCLK		6
+#define R9A07G044_GIC600_GICCLK		7
+#define R9A07G044_IA55_CLK		8
+#define R9A07G044_IA55_PCLK		9
+#define R9A07G044_MHU_PCLK		10
+#define R9A07G044_SYC_CNT_CLK		11
+#define R9A07G044_DMAC_ACLK		12
+#define R9A07G044_DMAC_PCLK		13
+#define R9A07G044_OSTM0_PCLK		14
+#define R9A07G044_OSTM1_PCLK		15
+#define R9A07G044_OSTM2_PCLK		16
+#define R9A07G044_MTU_X_MCK_MTU3	17
+#define R9A07G044_POE3_CLKM_POE		18
+#define R9A07G044_GPT_PCLK		19
+#define R9A07G044_POEG_A_CLKP		20
+#define R9A07G044_POEG_B_CLKP		21
+#define R9A07G044_POEG_C_CLKP		22
+#define R9A07G044_POEG_D_CLKP		23
+#define R9A07G044_WDT0_PCLK		24
+#define R9A07G044_WDT0_CLK		25
+#define R9A07G044_WDT1_PCLK		26
+#define R9A07G044_WDT1_CLK		27
+#define R9A07G044_WDT2_PCLK		28
+#define R9A07G044_WDT2_CLK		29
+#define R9A07G044_SPI_CLK2		30
+#define R9A07G044_SPI_CLK		31
+#define R9A07G044_SDHI0_IMCLK		32
+#define R9A07G044_SDHI0_IMCLK2		33
+#define R9A07G044_SDHI0_CLK_HS		34
+#define R9A07G044_SDHI0_ACLK		35
+#define R9A07G044_SDHI1_IMCLK		36
+#define R9A07G044_SDHI1_IMCLK2		37
+#define R9A07G044_SDHI1_CLK_HS		38
+#define R9A07G044_SDHI1_ACLK		39
+#define R9A07G044_GPU_CLK		40
+#define R9A07G044_GPU_AXI_CLK		41
+#define R9A07G044_GPU_ACE_CLK		42
+#define R9A07G044_ISU_ACLK		43
+#define R9A07G044_ISU_PCLK		44
+#define R9A07G044_H264_CLK_A		45
+#define R9A07G044_H264_CLK_P		46
+#define R9A07G044_CRU_SYSCLK		47
+#define R9A07G044_CRU_VCLK		48
+#define R9A07G044_CRU_PCLK		49
+#define R9A07G044_CRU_ACLK		50
+#define R9A07G044_MIPI_DSI_PLLCLK	51
+#define R9A07G044_MIPI_DSI_SYSCLK	52
+#define R9A07G044_MIPI_DSI_ACLK		53
+#define R9A07G044_MIPI_DSI_PCLK		54
+#define R9A07G044_MIPI_DSI_VCLK		55
+#define R9A07G044_MIPI_DSI_LPCLK	56
+#define R9A07G044_LCDC_CLK_A		57
+#define R9A07G044_LCDC_CLK_P		58
+#define R9A07G044_LCDC_CLK_D		59
+#define R9A07G044_SSI0_PCLK2		60
+#define R9A07G044_SSI0_PCLK_SFR		61
+#define R9A07G044_SSI1_PCLK2		62
+#define R9A07G044_SSI1_PCLK_SFR		63
+#define R9A07G044_SSI2_PCLK2		64
+#define R9A07G044_SSI2_PCLK_SFR		65
+#define R9A07G044_SSI3_PCLK2		66
+#define R9A07G044_SSI3_PCLK_SFR		67
+#define R9A07G044_SRC_CLKP		68
+#define R9A07G044_USB_U2H0_HCLK		69
+#define R9A07G044_USB_U2H1_HCLK		70
+#define R9A07G044_USB_U2P_EXR_CPUCLK	71
+#define R9A07G044_USB_PCLK		72
+#define R9A07G044_ETH0_CLK_AXI		73
+#define R9A07G044_ETH0_CLK_CHI		74
+#define R9A07G044_ETH1_CLK_AXI		75
+#define R9A07G044_ETH1_CLK_CHI		76
+#define R9A07G044_I2C0_PCLK		77
+#define R9A07G044_I2C1_PCLK		78
+#define R9A07G044_I2C2_PCLK		79
+#define R9A07G044_I2C3_PCLK		80
+#define R9A07G044_SCIF0_CLK_PCK		81
+#define R9A07G044_SCIF1_CLK_PCK		82
+#define R9A07G044_SCIF2_CLK_PCK		83
+#define R9A07G044_SCIF3_CLK_PCK		84
+#define R9A07G044_SCIF4_CLK_PCK		85
+#define R9A07G044_SCI0_CLKP		86
+#define R9A07G044_SCI1_CLKP		87
+#define R9A07G044_IRDA_CLKP		88
+#define R9A07G044_RSPI0_CLKB		89
+#define R9A07G044_RSPI1_CLKB		90
+#define R9A07G044_RSPI2_CLKB		91
+#define R9A07G044_CANFD_PCLK		92
+#define R9A07G044_GPIO_HCLK		93
+#define R9A07G044_ADC_ADCLK		94
+#define R9A07G044_ADC_PCLK		95
+#define R9A07G044_TSU_PCLK		96
 
-#define R9A07G044L_CLK_GIC600		0
-#define R9A07G044L_CLK_IA55		1
-#define R9A07G044L_CLK_SYC		2
-#define R9A07G044L_CLK_DMAC		3
-#define R9A07G044L_CLK_SYSC		4
-#define R9A07G044L_CLK_MTU		5
-#define R9A07G044L_CLK_GPT		6
-#define R9A07G044L_CLK_ETH0		7
-#define R9A07G044L_CLK_ETH1		8
-#define R9A07G044L_CLK_I2C0		9
-#define R9A07G044L_CLK_I2C1		10
-#define R9A07G044L_CLK_I2C2		11
-#define R9A07G044L_CLK_I2C3		12
-#define R9A07G044L_CLK_SCIF0		13
-#define R9A07G044L_CLK_SCIF1		14
-#define R9A07G044L_CLK_SCIF2		15
-#define R9A07G044L_CLK_SCIF3		16
-#define R9A07G044L_CLK_SCIF4		17
-#define R9A07G044L_CLK_SCI0		18
-#define R9A07G044L_CLK_SCI1		19
-#define R9A07G044L_CLK_GPIO		20
-#define R9A07G044L_CLK_SDHI0		21
-#define R9A07G044L_CLK_SDHI1		22
-#define R9A07G044L_CLK_USB0		23
-#define R9A07G044L_CLK_USB1		24
-#define R9A07G044L_CLK_CANFD		25
-#define R9A07G044L_CLK_SSI0		26
-#define R9A07G044L_CLK_SSI1		27
-#define R9A07G044L_CLK_SSI2		28
-#define R9A07G044L_CLK_SSI3		29
-#define R9A07G044L_CLK_MHU		30
-#define R9A07G044L_CLK_OSTM0		31
-#define R9A07G044L_CLK_OSTM1		32
-#define R9A07G044L_CLK_OSTM2		33
-#define R9A07G044L_CLK_WDT0		34
-#define R9A07G044L_CLK_WDT1		35
-#define R9A07G044L_CLK_WDT2		36
-#define R9A07G044L_CLK_WDT_PON		37
-#define R9A07G044L_CLK_GPU		38
-#define R9A07G044L_CLK_ISU		39
-#define R9A07G044L_CLK_H264		40
-#define R9A07G044L_CLK_CRU		41
-#define R9A07G044L_CLK_MIPI_DSI		42
-#define R9A07G044L_CLK_LCDC		43
-#define R9A07G044L_CLK_SRC		44
-#define R9A07G044L_CLK_RSPI0		45
-#define R9A07G044L_CLK_RSPI1		46
-#define R9A07G044L_CLK_RSPI2		47
-#define R9A07G044L_CLK_ADC		48
-#define R9A07G044L_CLK_TSU_PCLK		49
-#define R9A07G044L_CLK_SPI		50
+/* R9A07G044 Resets */
+#define R9A07G044_CA55_RST_1_0		0
+#define R9A07G044_CA55_RST_1_1		1
+#define R9A07G044_CA55_RST_3_0		2
+#define R9A07G044_CA55_RST_3_1		3
+#define R9A07G044_CA55_RST_4		4
+#define R9A07G044_CA55_RST_5		5
+#define R9A07G044_CA55_RST_6		6
+#define R9A07G044_CA55_RST_7		7
+#define R9A07G044_CA55_RST_8		8
+#define R9A07G044_CA55_RST_9		9
+#define R9A07G044_CA55_RST_10		10
+#define R9A07G044_CA55_RST_11		11
+#define R9A07G044_CA55_RST_12		12
+#define R9A07G044_GIC600_GICRESET_N	13
+#define R9A07G044_GIC600_DBG_GICRESET_N	14
+#define R9A07G044_IA55_RESETN		15
+#define R9A07G044_MHU_RESETN		16
+#define R9A07G044_DMAC_ARESETN		17
+#define R9A07G044_DMAC_RST_ASYNC	18
+#define R9A07G044_SYC_RESETN		19
+#define R9A07G044_OSTM0_PRESETZ		20
+#define R9A07G044_OSTM1_PRESETZ		21
+#define R9A07G044_OSTM2_PRESETZ		22
+#define R9A07G044_MTU_X_PRESET_MTU3	23
+#define R9A07G044_POE3_RST_M_REG	24
+#define R9A07G044_GPT_RST_C		25
+#define R9A07G044_POEG_A_RST		26
+#define R9A07G044_POEG_B_RST		27
+#define R9A07G044_POEG_C_RST		28
+#define R9A07G044_POEG_D_RST		29
+#define R9A07G044_WDT0_PRESETN		30
+#define R9A07G044_WDT1_PRESETN		31
+#define R9A07G044_WDT2_PRESETN		32
+#define R9A07G044_SPI_RST		33
+#define R9A07G044_SDHI0_IXRST		34
+#define R9A07G044_SDHI1_IXRST		35
+#define R9A07G044_GPU_RESETN		36
+#define R9A07G044_GPU_AXI_RESETN	37
+#define R9A07G044_GPU_ACE_RESETN	38
+#define R9A07G044_ISU_ARESETN		39
+#define R9A07G044_ISU_PRESETN		40
+#define R9A07G044_H264_X_RESET_VCP	41
+#define R9A07G044_H264_CP_PRESET_P	42
+#define R9A07G044_CRU_CMN_RSTB		43
+#define R9A07G044_CRU_PRESETN		44
+#define R9A07G044_CRU_ARESETN		45
+#define R9A07G044_MIPI_DSI_CMN_RSTB	46
+#define R9A07G044_MIPI_DSI_ARESET_N	47
+#define R9A07G044_MIPI_DSI_PRESET_N	48
+#define R9A07G044_LCDC_RESET_N		49
+#define R9A07G044_SSI0_RST_M2_REG	50
+#define R9A07G044_SSI1_RST_M2_REG	51
+#define R9A07G044_SSI2_RST_M2_REG	52
+#define R9A07G044_SSI3_RST_M2_REG	53
+#define R9A07G044_SRC_RST		54
+#define R9A07G044_USB_U2H0_HRESETN	55
+#define R9A07G044_USB_U2H1_HRESETN	56
+#define R9A07G044_USB_U2P_EXL_SYSRST	57
+#define R9A07G044_USB_PRESETN		58
+#define R9A07G044_ETH0_RST_HW_N		59
+#define R9A07G044_ETH1_RST_HW_N		60
+#define R9A07G044_I2C0_MRST		61
+#define R9A07G044_I2C1_MRST		62
+#define R9A07G044_I2C2_MRST		63
+#define R9A07G044_I2C3_MRST		64
+#define R9A07G044_SCIF0_RST_SYSTEM_N	65
+#define R9A07G044_SCIF1_RST_SYSTEM_N	66
+#define R9A07G044_SCIF2_RST_SYSTEM_N	67
+#define R9A07G044_SCIF3_RST_SYSTEM_N	68
+#define R9A07G044_SCIF4_RST_SYSTEM_N	69
+#define R9A07G044_SCI0_RST		70
+#define R9A07G044_SCI1_RST		71
+#define R9A07G044_IRDA_RST		72
+#define R9A07G044_RSPI0_RST		73
+#define R9A07G044_RSPI1_RST		74
+#define R9A07G044_RSPI2_RST		75
+#define R9A07G044_CANFD_RSTP_N		76
+#define R9A07G044_CANFD_RSTC_N		77
+#define R9A07G044_GPIO_RSTN		78
+#define R9A07G044_GPIO_PORT_RESETN	79
+#define R9A07G044_GPIO_SPARE_RESETN	80
+#define R9A07G044_ADC_PRESETN		81
+#define R9A07G044_ADC_ADRST_N		82
+#define R9A07G044_TSU_PRESETN		83
 
-#endif /* __DT_BINDINGS_CLOCK_R9A07G044L_CPG_H__ */
+#endif /* __DT_BINDINGS_CLOCK_R9A07G044_CPG_H__ */
diff --git a/include/dt-bindings/pinctrl/rzg2l-pinctrl.h b/include/dt-bindings/pinctrl/rzg2l-pinctrl.h
index c69a64b06a..dcc144535b 100644
--- a/include/dt-bindings/pinctrl/rzg2l-pinctrl.h
+++ b/include/dt-bindings/pinctrl/rzg2l-pinctrl.h
@@ -1,13 +1,26 @@
-/* SPDX-License-Identifier: GPL-2.0 */
+/* SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause) */
 /*
- * Defines macros and constants for Renesas RZ/G2L pin controller pin
- * muxing functions.
+ * This header provides constants for Renesas RZ/G2L family pinctrl bindings.
+ *
+ * Copyright (C) 2023 Renesas Electronics Corp.
+ *
  */
+
 #ifndef __DT_BINDINGS_RZG2L_PINCTRL_H
 #define __DT_BINDINGS_RZG2L_PINCTRL_H
 
 #define RZG2L_PINS_PER_PORT	8
 
+/*
+ * Store the pin index from its port and position number in bits[15-0].
+ * And store its peripheral function mode identifier in 3 bits [18-16].
+ * This macro is used in Linux device drivers
+ */
+#define RZG2L_PORT_PINMUX(b, p, f)	((b) * RZG2L_PINS_PER_PORT + (p) | ((f) << 16))
+
+/* Convert a port and pin label to its global pin index */
+ #define RZG2L_GPIO(port, pin)	((port) * RZG2L_PINS_PER_PORT + (pin))
+
 /*
  * Store the pin index from its port and position number in bits[11-0].
  * And store its peripheral function mode identifier in 3 bits [14-12]
-- 
2.25.1

